<!DOCTYPE html>

<html lang="en" xml:lang="en">

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="UM Bioinformatics Core Workshop Team" />


<title>Intro to Functional Analysis</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="shortcut icon" href="favicon-16x16.png" />
<link rel="manifest" href="/site.webmanifest">
<link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5">
<meta name="msapplication-TileColor" content="#da532c">
<meta name="theme-color" content="#ffffff">

<!--Favicon from https://favicon.io/emoji-favicons -->
<meta property="og:url" content="https://michmed.org/WXdGq"/>
<meta property="og:title" content="Intro to Functional Analysis"/>
<meta property="og:description" content="Functional Enrichment Analysis places a list of differentially expressed genes into a biological context. It presents a systems-level perspective that can reveal expression patterns, elucidate molecular mechanisms, and guide follow-up experiments." />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intro Functional Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="workshop-intro.html">Intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-overview.pdf">Functional analysis and ORA</a>
    </li>
    <li>
      <a href="02-IntroToWebGestaltandORA.html">ORA with WebGestalt</a>
    </li>
    <li>
      <a href="03-WebGestaltRORA.html">ORA with R</a>
    </li>
    <li>
      <a href="07A-advanced-visualizations.html">Advanced visualizations I</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-gsea-overview.pdf">Functional Analysis and GSEA</a>
    </li>
    <li>
      <a href="06-WebGestaltRGSEA.html">GSEA with R</a>
    </li>
    <li>
      <a href="gene-set-references.pdf">GO, KEGG, and friends</a>
    </li>
    <li>
      <a href="07B-advanced-visualizations.html">Advanced visualizations II</a>
    </li>
    <li>
      <a href="08-analysis-summary.html">Analysis summary</a>
    </li>
  </ul>
</li>
<li>
  <a href="workshop-wrap-up.html">Wrap up</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Intro to Functional Analysis</h1>
<h4 class="author">UM Bioinformatics Core Workshop Team</h4>

</div>


<style type="text/css">

body, td {
   font-size: 18px;
}
</style>
<script>
  addClassKlippyTo("pre.r, pre.markdown, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<div id="workshop-analysis-scripts" class="section level1">
<h1>Workshop analysis scripts</h1>
<div id="day-1" class="section level2">
<h2>Day 1</h2>
<pre class="r"><code># =========================================================================
# Day 1 - Getting Started with Functional enrichment
# =========================================================================

# -------------------------------------------------------------------------
# Get current working directory
getwd()

# -------------------------------------------------------------------------
# Set current working directory
setwd(&#39;~/IFUN_R&#39;)

# -------------------------------------------------------------------------
# Create directory structure
dir.create(&#39;scripts&#39;, recursive = TRUE, showWarnings = FALSE)
dir.create(&#39;results/figures&#39;, recursive = TRUE, showWarnings = FALSE)
dir.create(&#39;results/tables&#39;, recursive = TRUE, showWarnings = FALSE)
dir.create(&#39;results/rdata&#39;, recursive = TRUE, showWarnings = FALSE)

# =========================================================================
# ORA with WebGestlatR
# =========================================================================

# -------------------------------------------------------------------------
# Load the libraries
library(WebGestaltR)
library(tidyverse)

# Look at the manual for the next package/function name
?WebGestaltR

# List available organisms
listOrganism()

# -------------------------------------------------------------------------
# List available reference genesets for mmusculus
listGeneSet(organism = &#39;mmusculus&#39;) %&gt;% head(10)

# -------------------------------------------------------------------------
# Read diffex results
rsd_diffex = read_csv(&#39;inputs/bulk_de_results/de_deficient_vs_control_annotated.csv&#39;)
head(rsd_diffex)

# -------------------------------------------------------------------------
# Filter out the NAs
rsd_diffex = rsd_diffex %&gt;% dplyr::filter(!is.na(symbol))

# Verify we have fewer by noting dimension of resulting table
rsd_diffex

# -------------------------------------------------------------------------
# Pull all genes tested
rsd_ref_ora = rsd_diffex %&gt;% pull(symbol)

# -------------------------------------------------------------------------
# Preview the vector
head(rsd_ref_ora)

# -------------------------------------------------------------------------
# Pull diffex genes
rsd_sig_ora = rsd_diffex %&gt;%
  dplyr::filter(padj &lt; 0.05 &amp; abs(log2FoldChange) &gt; log2(1.5)) %&gt;%
  pull(symbol)

# -------------------------------------------------------------------------
# Preview the vector
head(rsd_sig_ora)

# -------------------------------------------------------------------------
# Write out list of DE interest genes to file
write_lines(rsd_sig_ora, file=&quot;./results/tables/deficient_DE_GeneList.txt&quot;)

# -------------------------------------------------------------------------
# Write out custom reference/background set of genes to file
write_lines(rsd_ref_ora, file=&quot;./results/tables/deficient_reference_GeneList.txt&quot;)

# -------------------------------------------------------------------------
# ORA on bulk RNA-seq DESeq2 results with GO - Biological Process knowledge base
rsd_ora_result = WebGestaltR(
  enrichMethod = &#39;ORA&#39;,
  organism = &#39;mmusculus&#39;,
  enrichDatabase = c(&#39;geneontology_Biological_Process_noRedundant&#39;),
  interestGene = rsd_sig_ora,
  referenceGene = rsd_ref_ora,
  interestGeneType = &#39;genesymbol&#39;,
  referenceGeneType = &#39;genesymbol&#39;,
  fdrThr = 0.1,
  outputDirectory = &#39;./results&#39;,
  projectName = &#39;deficient_vs_control_ORA-GO_BP&#39;)

# -------------------------------------------------------------------------
# View the first few results
head(rsd_ora_result)


# =========================================================================
# Advanced Visualizations I
# =========================================================================
# Load addtional package
library(ComplexHeatmap)

# -------------------------------------------------------------------------
# OPTIONAL - Read in results from file
# rsd_ora_result = read_delim(&#39;results/Project_deficient_vs_control_ORA_GO_BP/enrichment_results_deficient_vs_control_ORA_GO_BP.txt&#39;)

# look at our data
head(rsd_ora_result)

# make a copy to match expected name for ORA
enriched_ora &lt;- rsd_ora_result
head(enriched_ora)

# -------------------------------------------------------------------------
# create empty list
go_geneSets &lt;- list()

# -------------------------------------------------------------------------
# create lists of overlapping DE genes for each GO term
for(i in 1:length(enriched_ora$description)){
  go_name &lt;- enriched_ora$description[i] # descriptive GO term names
  go_genes &lt;- str_split_1(enriched_ora$userId[i], &quot;;&quot;) # list of gene symbols from table entry
  go_geneSets[[go_name]] &lt;- c(go_genes)
}

# -------------------------------------------------------------------------
# check the named lists
head(go_geneSets)

# -------------------------------------------------------------------------
# check what genes are shared between the two terms
intersect(go_geneSets$`neuron death`, go_geneSets$`intrinsic apoptotic signaling pathway`)

# -------------------------------------------------------------------------
# First make the combination matrix from our list of genes for each GO term
m = make_comb_mat(go_geneSets)
m

# -------------------------------------------------------------------------
# Make simple UpSet plot
UpSet(m)


# -------------------------------------------------------------------------
# Make altnerative UpSet plot
upsetGO &lt;- UpSet(m, comb_order = rev(order(comb_size(m))), lwd = 3,
                 comb_col = c(&quot;black&quot;, &quot;blue&quot;, &quot;gold3&quot;)[comb_degree(m)],
                 left_annotation = upset_left_annotation(m, add_numbers = TRUE))
upsetGO


# -------------------------------------------------------------------------
# Output UpSet plot to file
png(file = &quot;./results/figures/deficient_vs_control_GO-BP_UpSetPlot.png&quot;, width = 1000, height = 400)
draw(upsetGO) # draw the plot
dev.off() # Close the graphics device</code></pre>
</div>
<div id="day-2" class="section level2">
<h2>Day 2</h2>
<pre class="r"><code># =========================================================================
# GSEA with WebGestlatR
# =========================================================================

# -------------------------------------------------------------------------
# Set our working directory
setwd(&quot;~/IFUN_R&quot;) # can use arrow in console to quickly see working directory in file pane

# -------------------------------------------------------------------------
# Load the libraries
library(WebGestaltR)
library(tidyverse)

# -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D21 vs D7 in pericyte cluster
isc_d21_v_d7 = read_csv(&#39;inputs/single_cell_de_results/de_pseudo_pericyte_D21_vs_D7.csv&#39;)
# -------------------------------------------------------------------------
# Preview the result
head(isc_d21_v_d7)

# -------------------------------------------------------------------------
# Check how many genes were DE based on original thresholds
table(isc_d21_v_d7$p_val_adj &lt; 0.05 &amp; abs(isc_d21_v_d7$avg_log2FC) &gt; 1.5)


# -------------------------------------------------------------------------
# How many symbols are NA?
table(is.na(isc_d21_v_d7$gene))


# -------------------------------------------------------------------------
# Select gene and avg_log2FC columns
isc_d21_v_d7_gsea = isc_d21_v_d7 %&gt;%
  dplyr::select(gene, avg_log2FC)

# -------------------------------------------------------------------------
# Preview the table
head(isc_d21_v_d7_gsea)

# -------------------------------------------------------------------------
# Run GSEA
isc_d21_v_d7_gsea_result = WebGestaltR(
  enrichMethod = &#39;GSEA&#39;,
  nThreads = 8,
  organism = &#39;mmusculus&#39;,
  enrichDatabase = c(&#39;pathway_KEGG&#39;),
  interestGene = isc_d21_v_d7_gsea,
  interestGeneType = &#39;genesymbol&#39;,
  fdrThr = 0.1,
  outputDirectory = &#39;./results&#39;,
  projectName = &#39;D21_v_D7_GSEA_KEGG&#39;, 
  cache = NULL)

# clean up session
gc()

# -------------------------------------------------------------------------
# View the first few results
head(isc_d21_v_d7_gsea_result)

# --------------------------
# make it easier to view
View(isc_d21_v_d7_gsea_result)

# - look at second comparison

# -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D7 vs D7 in pericyte cluster
isc_d7_v_d0 = read_csv(&#39;inputs/single_cell_de_results/de_pseudo_pericyte_D7_vs_D0.csv&#39;)

# -------------------------------------------------------------------------
# Preview the result
head(isc_d7_v_d0)


# -------------------------------------------------------------------------
# Select gene and avg_log2FC columns
isc_d7_v_d0_gsea = isc_d7_v_d0 %&gt;%
  dplyr::select(gene, avg_log2FC)

# -------------------------------------------------------------------------
# Preview the table
head(isc_d7_v_d0_gsea)

# -------------------------------------------------------------------------
# Select gene and avg_log2FC columns
isc_d7_v_d0_gsea = isc_d7_v_d0 %&gt;%
  dplyr::select(gene, avg_log2FC)

# -------------------------------------------------------------------------
# Preview the table
head(isc_d7_v_d0_gsea)


# -------------------------------------------------------------------------
# Run GSEA for Day 7 vs Day 0 (EDITED to add random seed)
set.seed(12345)  # set random seed first

isc_d7_v_d0_gsea_result = WebGestaltR(
  enrichMethod = &#39;GSEA&#39;,
  nThreads = 8,
  organism = &#39;mmusculus&#39;,
  enrichDatabase = c(&#39;pathway_KEGG&#39;),
  interestGene = isc_d7_v_d0_gsea,
  interestGeneType = &#39;genesymbol&#39;,
  fdrThr = 0.1,
  outputDirectory = &#39;./results&#39;,
  projectName = &#39;D7_v_D0_GSEA_KEGG&#39;, 
  cache = NULL)

# clean up session
gc()

# -------------------------------------------------------------------------
# View the results for the other timepoint
View(isc_d7_v_d0_gsea_result)

# =========================================================================
# Advanced Visualizations II
# =========================================================================

# -------------------------------------------------------------------------
# Load additional libraries
library(ComplexHeatmap)
library(ggVennDiagram)

# -------------------------------------------------------------------------
# Check current working directory
getwd()

# -------------------------------------------------------------------------
# Create list of enriched KEGG pathways 
KEGG_results &lt;- c()
KEGG_results[[&quot;d7_v_d0&quot;]] = isc_d7_v_d0_gsea_result$description
KEGG_results[[&quot;d21_v_d7&quot;]] = isc_d21_v_d7_gsea_result$description
head(KEGG_results)

# -------------------------------------------------------------------------
# Identify total number of unique pathways enriched across both timepoints
union(KEGG_results[[&quot;d7_v_d0&quot;]], KEGG_results[[&quot;d21_v_d7&quot;]]) %&gt;% length()

# -------------------------------------------------------------------------
# Identify number of pathways shared between timepoints
intersect(KEGG_results[[&quot;d7_v_d0&quot;]], KEGG_results[[&quot;d21_v_d7&quot;]]) %&gt;% length()

# -------------------------------------------------------------------------
# Make simple plot to compare KEGG pathways enriched for each timepoint
KEGG_venn &lt;- ggVennDiagram(KEGG_results, label_alpha = 0, 
                           label_size = 6, set_size = 6,
) +
  ggplot2::scale_fill_gradient(low=&quot;royalblue&quot;,high = &quot;yellow&quot;) +
  coord_flip()
KEGG_venn

# -------------------------------------------------------------------------
# Output UpSet plot to file
png(file = &quot;./results/figures/Pericytes_KEGG_vennDiagram.png&quot;, width = 600, height = 400)
KEGG_venn # draw the plot
dev.off() # Close the graphics device



# -------------------------------------------------------------------------
# Create &#39;tidy&#39; version  of Day 7 vs Day 0 GSEA results
isc_d7_v_d0_gsea_simplified &lt;- isc_d7_v_d0_gsea_result %&gt;% 
  select(geneSet, description, normalizedEnrichmentScore, size, FDR) %&gt;% 
  pivot_longer(cols = !c(geneSet, description, FDR, size)) %&gt;% 
  mutate(timepoint = &quot;D7_v_D0&quot;, .before = 1)

# -------------------------------------------------------------------------
# Create &#39;tidy&#39; version  of Day 21 vs Day 7 GSEA results
isc_d21_v_d7_gsea_simplified &lt;- isc_d21_v_d7_gsea_result %&gt;% 
  select(geneSet, description, normalizedEnrichmentScore, size, FDR) %&gt;% 
  pivot_longer(cols = !c(geneSet, description, FDR, size)) %&gt;% 
  mutate(timepoint = &quot;D21_v_D7&quot;, .before = 1)

View(isc_d21_v_d7_gsea_simplified)

# -------------------------------------------------------------------------
# Combine &#39;tidy&#39; results into single table
KEGG_results_combined &lt;- rbind(isc_d7_v_d0_gsea_simplified, isc_d21_v_d7_gsea_simplified) 
KEGG_results_combined &lt;- KEGG_results_combined %&gt;% 
  mutate(timepoint = factor(timepoint, levels=c(&quot;D7_v_D0&quot;, &quot;D21_v_D7&quot;)),
         direction = factor(ifelse(value &lt; 0, &quot;negative NES&quot;, &quot;positive NES&quot;)),
         &quot;positive NES&quot;, &quot;negative NES&quot;) # fix levels &amp; add column for direction

# -------------------------------------------------------------------------
# Check table
str(KEGG_results_combined)


# -------------------------------------------------------------------------
# Graph simple KEGG pathway enrichments for both timepoints
KEGG_comparison &lt;- ggplot(KEGG_results_combined, 
                          aes(y=value, x=reorder(description, value))) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;) +
  facet_wrap(~timepoint) +
  xlab(&quot;&quot;) + coord_flip()
KEGG_comparison


# -------------------------------------------------------------------------
# Subset to results that pass more stringent results
KEGG_results_strict &lt;- KEGG_results_combined %&gt;% filter(FDR &lt; 0.025 &amp; size &gt; 60)

# -------------------------------------------------------------------------
# Graph simple KEGG pathway enrichments for both timepoints
KEGG_comparison &lt;- ggplot(KEGG_results_strict, aes(y=value, x=reorder(description, value))) + 
  geom_segment(aes(y = 0, x = description, yend = value, xend = description),
               color = ifelse(KEGG_results_strict$direction %in% c(&quot;positive NES&quot;), &quot;red&quot;, &quot;blue&quot;),
               linewidth = 1) +
  geom_point(aes(color=FDR, size = size), stat=&quot;identity&quot;) +
  scale_color_continuous(low = &quot;black&quot;, high = &quot;grey&quot;) +
  facet_wrap(~timepoint) +
  geom_hline(yintercept=0, linetype=&quot;dashed&quot;, color = &quot;lightgrey&quot;) +
  labs(title=&quot;KEGG enrichments in pericytes&quot;) +
  xlab(&quot;&quot;) + theme_bw() +
  coord_flip()
KEGG_comparison


# -------------------------------------------------------------------------
# Output Lollipop plot to file
png(file = &quot;./results/figures/Pericytes_KEGG_lollipopComparison.png&quot;, width = 700, height = 600, res = 100)
KEGG_comparison # draw the plot
dev.off() # Close the graphics device</code></pre>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
