<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Introduction to WebGestalt and ORA</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link rel="shortcut icon" href="favicon-16x16.png" />
<!--
Favicon from built via
https://favicon.io/emoji-favicons/computer/
-->
<meta name="google-site-verification" content="t-Wgs-RXt8xi94RSoFamxoD2P2h-Hh7M_t_iOuxFZUo" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intro to Functional Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="workshop-intro.html">Intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="introduction.html">intro</a>
    </li>
    <li>
      <a href="more-stuff.html">more stuff</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="fancy-stuff.html">Fancy Stuff</a>
    </li>
  </ul>
</li>
<li>
  <a href="workshop-wrap-up.html">Wrap up</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Introduction to WebGestalt and ORA</h1>

</div>


<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>
<script>
  addClassKlippyTo("pre.r, pre.markdown, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<pre><code>## [1] &quot;/home/workshop/damki/workshop-intro-functional-analysis/source&quot;</code></pre>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<ul>
<li>1</li>
<li>2</li>
<li>3</li>
</ul>
<p>…</p>
</div>
<div id="webgestalt-overview" class="section level1">
<h1>WebGestalt Overview</h1>
<p>We’ve discussed some of the motivations for performing functional
enrichment analysis, but what tools can we use to perform these kinds of
analyses?</p>
<p>One option is <a href="http://www.webgestalt.org/">WebGestalt</a>
(WEB-based GEne SeT AnaLysis Toolkit) which has a web-based interface
for running functional enrichment analysis. It supports a wide range of
organisms and offers several methods for enrichment analysis.</p>
<div
id="overrepresentation-results-for-bulk-rna-seq-data-using-webgestalt"
class="section level2">
<h2>Overrepresentation results for bulk RNA-seq data using
WebGestalt</h2>
<!-- per group discussions - plan to move this section to the start to give an overview of the results before using the tool -->
<p>Let’s look through an example of ORA (Over Representation Analysis)
report from WebGestalt, similar to what might be shared by a
collaborator. These results were generated for our bulk RNA-seq data,
which compared gene expression between
<!-- what was in the input tissue??--> wild-type mice fed a iron
deficient vs control diet.</p>
<!-- TO ADD - figure of experimental design from RSD -->
<p>We’ll go through the steps for how to run ORA using the browser
version together shortly but let’s first look through what WebGestalt
reports and what results we can expect.</p>
<p><br></p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_d_ORA-jobInfo.png"
alt="WebGestalt ORA example" /><br />
At the top of the report, WebGestalt summarizes what was submitted. The
bulk RNA-seq had 189 DE genes in total but only 180 genes mapped to gene
symbols of the tool’s reference (<code>entrezgene</code>) and then only
125 genes were annotated to the functional categories selected for this
analysis, which was
<code>geneontology_Biological_Process_noRedundant</code>.</p>
<p>We can also see that the reference list was user provided
(<code>upload/deficient_background_GeneList_XXXXXX.txt</code>) and that
14992 and 9375 were mapped to entrezgene gene symbols and functional
catagories, respectively.</p>
<p><br></p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_f_ORA-barplot-results.png"
alt="WebGestalt ORA example" /><br />
Next, we can see the top 10 enriched GO-terms for our submitted DE genes
from the non-redundant Biological Process GO-term set. While 10 terms
are reported, only 4 are significant after FDR multiple hypothesis
correction. <!-- add more detail here?? --></p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_f_ORA-table-results.png"
alt="WebGestalt ORA example" /></p>
<p>Next, we can see the top 10 enriched GO-terms for our submitted DE
genes from the non-redundant Biological Process GO-term set. While 10
terms are reported, only 4 are significant after FDR multiple hypothesis
correction.</p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_f_ORA-DAG-results.png"
alt="WebGestalt ORA example" /></p>
<p>If we look at the DAG map of GO terms included in the report, we can
see that some of the significant terms are related, like
<code>intrinsic apoptotic signaling pathway</code> and
<code>neuron apoptotic process</code>. This can be useful for
interpreting why a GO term that’s unlikely to be applicable to the
experiment (e.g. neuronal process from non-brain tissue) might be coming
up as enriched.</p>
<!-- change to dropdown? -->
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_f_ORA-volcano-results.png"
alt="WebGestalt ORA example" /></p>
<div id="individual-go-term-examples" class="section level3">
<h3>Individual GO-term examples</h3>
<p>Next let’s look at some of the significantly enriched terms.</p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_g_ORA-Glut-enrichment.png"
alt="WebGestalt ORA Glutathione metabolic process" /></p>
<p>Starting with glutathione metabolic process we see …
<!-- add more context --></p>
<p>Based on the experimental design, is finding a enrichment for a
metabolic process surprising? Is it surprising to find that this
specific process is enriched?
<!--- add call out box or quote to highlight question ---></p>
<p>Let’s look at some of our other enriched results.</p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_gg_ORA-Apoptosis-enrichment.png"
alt="WebGestalt ORA intrinsic apoptotic signaling pathway" /></p>
<p>For the intristic apoptotic signaling pathway, we see …
<!-- add more context --></p>
<p>For the neuron apoptotic process, would we expect to see similar or
difference genes represented for that GO term annotation&gt;?</p>
<!-- add example of neuronal enrichment to show overlap in genes between terms -->
<!-- transition back to interface and showing how to submit data -->
<p>Next, let’s go back to look at the <a
href="http://www.webgestalt.org/">WebGestalt interface</a> and
familiarize ourselves with the submission process and documentation.</p>
</div>
</div>
</div>
<div id="webgestalt-interface" class="section level1">
<h1>WebGestalt interface</h1>
<p>If we look at the <a href="http://www.webgestalt.org/">WebGestalt
homepage</a>, we can see that it has several sections:</p>
<!-- map abstract schematics/inputs to WebGestalt 
1) approach
2) database
3) query genes / background -->
<p><img
src="images/content/02-WebGestalt_browser/02-WebGestalt_topOfPage.png"
alt="WebGestalt homepage top" /> <br></p>
<ul>
<li>At the top, we have the main navigation menu, which includes links
including to the Manual, Citation, a User Forum, and a link to the 2019
version of the tool.</li>
<li>Then, the main section is the “Basic parameters” box, which allows
us to select the organism, analysis type, and database.</li>
</ul>
<p><br></p>
<p><img
src="images/content/02-WebGestalt_browser/02-WebGestalt_inputInset.png"
alt="WebGestalt homepage inputs" /><br />
<br></p>
<ul>
<li><p>Within the basic parameters box, there is a smaller box that
looks like its for specifying the input data, which we’ll circle back to
shortly.</p></li>
<li><p>Below that, there is a area labeled “Advanced parameters”, which
we’ll skip for now, but that includes options for adjusting the analysis
parameters, such as the statistical method and multiple testing
correction.</p></li>
</ul>
<p><br></p>
<div id="webgestalt-documentation" class="section level2">
<h2>WebGestalt documentation</h2>
<p><img
src="images/content/02-WebGestalt_browser/02-webgestalt_ClickManual.png"
alt="WebGestalt homepage find manual" /></p>
<p>A good practice when using a new tool is to read the documentation
provided by the developers, which we can find by selecting Manual from
the top of WebGestalt site.</p>
<p><img
src="images/content/02-WebGestalt_browser/02-webgestalt_TopOfManual.png"
alt="WebGestalt homepage manual top" /></p>
<p>At the top of the manual page, we can see an overview of some of the
related publications as well as a comparison between an older 2019
version and the current 2024 version.</p>
<p><img
src="images/content/02-WebGestalt_browser/02-webgestalt_Manual2ndPage.png"
alt="WebGestalt homepage manual page 2" /></p>
<p>If we scroll down, then we can see a bit of an introduction to the
tool. However, the manual seems to already have some terminology and
assumptions about a user’s familiarity with functional enrichment so
let’s go back to the tool site and focus on a more limited set of
options for our first pass.</p>
<!-- try to start with results downloaded for RSD DE data -->
</div>
</div>
<div id="webgestalt-analysis-demonstration" class="section level1">
<h1>WebGestalt analysis demonstration</h1>
<p>Let’s start with some of the basic parameters and delve into the
options we want to start with.</p>
<!-- Click through methods to show ORA vs other options & have demo data -->
<!-- Load demo data to show what inputs are needed-->
<!-- Note input gene list options - symbol, entrez, ensembl, etc. which will -->
<div id="input-data" class="section level2">
<h2>Input data</h2>
<p>Bulk RNA-seq results from our <a
href="https://umich-brcf-bioinf.github.io/workshop-rnaseq-demystified/main/html/">RNA-seq
demystified workshop</a> - statistics from comparision between deficient
vs control mice using DESeq2 (file =
<code>DE_results_deficient_vs_control_annotated.csv</code>)</p>
</div>
<div id="hands-on-review-input-data-and-prepare-for-analysis"
class="section level2">
<h2>Hands-on: Review input data and prepare for analysis</h2>
<!-- Consider moving this until after the WebGestalt introduction, so that the learners can see that the tool requires a list of DE vs non-DE for best results -->
<p>To start, we’ll load the input data and review it. The data is in CSV
format, so we can use <code>read_csv()</code> from the
<code>tidyverse</code> library to load it into R.</p>
<pre class="r"><code># Load necessary libraries
library(tidyverse)

# Check working directory
getwd()</code></pre>
<pre><code>[1] &quot;/home/workshop/damki/workshop-intro-functional-analysis/source&quot;</code></pre>
<pre class="r"><code># Load the bulk RNA-seq results
deficient.v.control_bulkDE &lt;- read_csv(&quot;../data/de_deficient_vs_control_annotated.csv&quot;) #TO REVISE - path may need to be adjusted

# Check the first few rows of the data
head(deficient.v.control_bulkDE)</code></pre>
<pre><code># A tibble: 6 × 9
  id                 symbol baseMean log2FoldChange lfcSE   stat pvalue   padj call 
  &lt;chr&gt;              &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;
1 ENSMUSG00000000001 Gnai3   1490.            0.278 0.148  1.88  0.0605  0.325 NS   
2 ENSMUSG00000000028 Cdc45   1749.            0.222 0.129  1.72  0.0853  0.386 NS   
3 ENSMUSG00000000031 H19     2152.            0.136 0.284  0.478 0.633   0.867 NS   
4 ENSMUSG00000000037 Scml2     24.9           0.600 0.562  1.07  0.286  NA     NS   
5 ENSMUSG00000000049 Apoh       7.78         -1.23  1.15  -1.07  0.285  NA     NS   
6 ENSMUSG00000000056 Narf   19654.           -0.201 0.167 -1.20  0.229   0.596 NS   </code></pre>
<p>In the table above, we have the following columns:</p>
<ul>
<li><code>id</code>: The ENSEMBL gene identifier.</li>
<li><code>symbol</code>: The gene symbol.</li>
<li><code>baseMean</code>: The average expression of the gene across all
samples.</li>
<li><code>log2FoldChange</code>: The log2 fold change in expression
between the deficient and control samples.</li>
<li><code>lfcSE</code>: The standard error of the log2 fold change.</li>
<li><code>stat</code>: The test statistic for the differential
expression test.</li>
<li><code>pvalue</code>: The p-value for the differential expression
test.</li>
</ul>
<p>A key attribute of this table is that includes statistics for all
genes included in the comparison, not just those that are differentially
expressed. This allows us to use functional enrichment tools that use
both a “positive” set of DE genes and a “background” set of all genes
measured in the experiment, like ORA.</p>
<p>Next, we’ll prepare the data for analysis by creating a list of DE
genes and a background set of all genes. We’ll use the
<code>log2FoldChange</code> and <code>pvalue</code> columns to identify
DE genes, and we’ll use the <code>gene</code> column as our gene
identifiers.</p>
<p><!-- include note about reproducibility and knowing thresholds --></p>
<pre class="r"><code># Generate positive set (DE gene symbols)
deficient_DE_only &lt;- deficient.v.control_bulkDE %&gt;%
  filter(padj &lt; 0.05 &amp; abs(log2FoldChange) &gt; log2(1.5))

deficient_DE_GeneList &lt;- deficient_DE_only %&gt;% pull(symbol) %&gt;% na.omit()

head(deficient_DE_only)</code></pre>
<pre><code># A tibble: 6 × 9
  id                 symbol baseMean log2FoldChange lfcSE  stat     pvalue     padj call 
  &lt;chr&gt;              &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;      &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;
1 ENSMUSG00000000275 Trim25   1662.          -0.674 0.200 -3.38 0.000734   0.0312   Down 
2 ENSMUSG00000000861 Bcl11a   1679.           0.685 0.210  3.27 0.00109    0.0398   Up   
3 ENSMUSG00000001281 Itgb7     532.           1.15  0.242  4.75 0.00000203 0.000554 Up   
4 ENSMUSG00000002109 Ddb2      125.           1.06  0.300  3.54 0.000398   0.0207   Up   
5 ENSMUSG00000002985 Apoe      157.          -1.16  0.345 -3.35 0.000814   0.0329   Down 
6 ENSMUSG00000003865 Gys1       89.3          2.26  0.627  3.61 0.000310   0.0180   Up   </code></pre>
<pre class="r"><code>head(deficient_DE_GeneList)</code></pre>
<pre><code>[1] &quot;Trim25&quot; &quot;Bcl11a&quot; &quot;Itgb7&quot;  &quot;Ddb2&quot;   &quot;Apoe&quot;   &quot;Gys1&quot;  </code></pre>
<pre class="r"><code># Repeat to generate background set (all gene names)
deficient_background_GeneList &lt;- deficient.v.control_bulkDE %&gt;% pull(symbol) %&gt;% na.omit()</code></pre>
<pre class="r"><code># write out to file (text only?)
write(deficient_DE_GeneList, file =&quot;../data/deficient_DE_GeneList.txt&quot;, ncolumns = 1)
write(deficient_background_GeneList, file =&quot;../data/deficient_background_GeneList.txt&quot;, ncolumns = 1)</code></pre>
<p>These outputs then allow us to submit data via the web browser
version of WebGestalt.</p>
<div id="running-webgestalt-with-our-bulk-rna-seq-results"
class="section level3">
<h3>Running WebGestalt with our bulk RNA-seq results</h3>
<p><br></p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_a_submission_withError.png"
alt="WebGestalt submit data for ORA" /></p>
<p><br></p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_b_submission_errorMessage.png"
alt="WebGestalt submit data for ORA" /></p>
<p><br></p>
<p>We ran into an error - why? Let’s go back to the submission page and
see if we can find the issue.</p>
<p><br></p>
<p><img
src="images/content/02-WebGestalt_browser/02-bulkRNA_a_submission_withErrorHighlighted.png"
alt="WebGestalt submit data for ORA" /><br />
If we look at the submission page, the
<code>ID type for uploaded reference list</code> is “Gene name”, which
doesn’t match the ID type we used for our gene list, which was “Gene
symbol”. Let’s change the
<code>ID type for uploaded reference list</code> to “Gene symbol” and
re-submit.</p>
<p><br></p>
<hr />
</div>
</div>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<ul>
<li>1</li>
<li>2</li>
<li>3</li>
</ul>
<br/> <br/>
<hr/>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"><a href="workshop-intro.html">Previous lesson</a></th>
<th align="center"><a href="#top">Top of this lesson</a></th>
<th align="right"><a href="more-stuff.html">Next lesson</a></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
