<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>ORA with R</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link rel="shortcut icon" href="favicon-16x16.png" />
<!--
Favicon from built via
https://favicon.io/emoji-favicons/computer/
-->
<meta name="google-site-verification" content="t-Wgs-RXt8xi94RSoFamxoD2P2h-Hh7M_t_iOuxFZUo" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intro Functional Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="workshop-intro.html">Intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-overview.pdf">Functional analysis and ORA</a>
    </li>
    <li>
      <a href="02-IntroToWebGestaltandORA.html">ORA with WebGestalt</a>
    </li>
    <li>
      <a href="03-WebGestaltRORA.html">ORA with R</a>
    </li>
    <li>
      <a href="07A-advanced-visualizations.html">Advanced visualizations I</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-gsea-overview.pdf">Functional Analysis and GSEA</a>
    </li>
    <li>
      <a href="06-WebGestaltRGSEA.html">GSEA with R</a>
    </li>
    <li>
      <a href="gene-set-references.pdf">GO, KEGG, and friends</a>
    </li>
    <li>
      <a href="07B-advanced-visualizations.html">Advanced visualizations II</a>
    </li>
    <li>
      <a href="08-analysis-summary.html">Analysis summary and next steps</a>
    </li>
  </ul>
</li>
<li>
  <a href="workshop-wrap-up.html">Wrap up</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ORA with R</h1>

</div>


<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>
<script>
  addClassKlippyTo("pre.r, pre.markdown, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<p><img src="images/wayfinder/wayfinder-03.png"
alt="workshop wayfinder" /> <br></p>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<ul>
<li>Generate the inputs for over-representation analysis (ORA) in
WebGestaltR from a differential expression analysis.</li>
<li>Use WebGestaltR to do an ORA analysis.</li>
<li>Understand how to interpret the results of ORA analysis.</li>
</ul>
</div>
<div id="getting-started" class="section level1">
<h1>Getting started</h1>
<p>As we have seen with the web interface for WebGestalt, an
over-representation analysis on differential expression results
requires:</p>
<ul>
<li>An <em>input</em> list of differentially expressed - subject to
cutoffs for significance - that will be queried.<br />
</li>
<li>A <em>reference/background gene set</em> of genes that are expected
to have be measured in the experiment</li>
<li>The <em>functional/biological knowledge database</em> which defines
the gene sets that will be used to test for over-representation.</li>
</ul>
<p>In addition to the browser version of <code>WebGestalt</code>, the
authors also created an <a
href="https://bzhanglab.github.io/WebGestaltR/">R package called
<code>WebGestaltR</code></a> that can be used to run the tool in R
instead of by using their webserver. This allows us to run functional
enrichments in a more reproducible manner since we can record the exact
inputs and parameters used.</p>
</div>
<div id="orienting-on-rstudio" class="section level1">
<h1>Orienting on RStudio</h1>
<p>To get started, let’s <strong>log in to the workshop server by going
here</strong>: <a href="https://bfx-workshop02.med.umich.edu"
target="blank">https://bfx-workshop02.med.umich.edu</a> <br></p>
<p>The login page for the server looks like:</p>
<p><img src="images/content/03-WebGestalt_R/01_rstudio_login_screen.png" alt="rstudio login screen" style="width: 1000px;"/>
<br></p>
<p>Enter your user credentials and click <kbd>Sign In</kbd>. The RStudio
interface should load and look something like:</p>
<p><img src="images/content/03-WebGestalt_R/02_rstudio_session_default.png" alt="rstudio default session" style="width:1000px;"/>
<br></p>
<p><strong>Checkpoint</strong> - after you’ve logged in put up a green
check; if you need help put up a red check and post in slack.</p>
<p>RStudio is an integrated development environment where you can write,
execute, and see the results of your code. The interface is arranged in
different panes:</p>
<ul>
<li>The <strong>Console pane</strong> along the left where you can enter
commands and execute them.</li>
<li>The <strong>Environment pane</strong> in the upper right shows any
variables you have created, along with their values.</li>
<li>The pane in the lower right has a few functions:
<ul>
<li>The <strong>Files tab</strong> let’s you navigate the file
system.</li>
<li>The <strong>Plots tab</strong> displays any plots from code run in
the Console.</li>
<li>The <strong>Help tab</strong> displays the documentation of
functions.</li>
</ul></li>
</ul>
<div id="commands-in-the-script" class="section level2">
<h2>Commands in the Script</h2>
<p>When first opening RStudio, a script file is not automatically
opened. We’ll create our script file to record our analysis and make it
more reproducible by clicking on the icon in the upper-left of the
interface (a blank piece of paper with a + sign), and selecting <KBD>R
Script</KBD>.</p>
<p><img src="images/content/03-WebGestalt_R/04_rstudio_create_script.png" alt="rstudio create script" style="width:1000px;"/>
<br></p>
<p>The new pane that opens is the <strong>Source pane</strong>:</p>
<p><img src="images/content/03-WebGestalt_R/05_rstudio_script_pane.png" alt="rstudio script pane" style="width:1000px;"/>
<br></p>
<p>Notice that if we press <KBD>Enter</KBD> in the Source pane, we get a
new line instead of running the code.</p>
<p>In order to execute code, we press <KBD>Ctrl + Enter</KBD> either on
the single line we want to run, or on a highlighted block of code. For
example if we type in the script file:</p>
<pre><code>3+2</code></pre>
<p>We then see that code executed, along with its result in in the
Console pane.</p>
<p><img src="images/content/03-WebGestlat_R/06_rstudio_script_execution.png" alt="rstudio script execution" style="width:1000px;"/></p>
</div>
<div id="summary-console-vs-script" class="section level2">
<h2>Summary: Console vs Script</h2>
<p>Here are some of the key differences between the Console and Script
panes in RStudio:</p>
<table>
<thead>
<tr class="header">
<th align="left">Console</th>
<th align="left">Script</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Ephemeral code</td>
<td align="left">Preserved code</td>
</tr>
<tr class="even">
<td align="left">Run with <KBD>Enter</KBD></td>
<td align="left">Run with <KBD>Ctrl + Enter</KBD></td>
</tr>
<tr class="odd">
<td align="left">Hard to share</td>
<td align="left">Easy to share</td>
</tr>
</tbody>
</table>
</div>
<div id="configuring-rstudio" class="section level2">
<h2>Configuring RStudio</h2>
<p>All of the panes in RStudio have configuration options. For example,
you can minimize/maximize a pane or resize panes by dragging the
borders. The most important customization options for pane layout are in
the <KBD>View</KBD> menu. Other options such as font sizes,
colors/themes, and more are in the <KBD>Tools</KBD> menu under
<KBD>Global Options</KBD>.</p>
<p>We can enable soft-wrapping of code by selecting <KBD>Code</KBD> and
then <KBD>Soft Wrap Long Lines</KBD>.</p>
<p><img src="images/07_rstudio_soft_wrap.png" alt="rstudio code execution" style="width:1000px;"/></p>
</div>
</div>
<div id="workshop-flow" class="section level1">
<h1>Workshop flow</h1>
<p>To accomodate learning styles and to keep us moving along, we’ll
provide code in three different ways, and you can get that code into
RStudio in corresponding ways:</p>
<table>
<colgroup>
<col width="48%" />
<col width="51%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Source of Code</th>
<th align="left">Execution of Code</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Zoom screen share</td>
<td align="left">Type the code yourself.</td>
</tr>
<tr class="even">
<td align="left">Slack</td>
<td align="left">Copy and paste code into RStudio.</td>
</tr>
<tr class="odd">
<td align="left">Website</td>
<td align="left">Use code block copy button and paste into RStudio.</td>
</tr>
</tbody>
</table>
<p><strong>Questions?</strong></p>
</div>
<div id="setting-up" class="section level1">
<h1>Setting up</h1>
<p>Before we begin, the folder structure of a working directory /
project organizes all the relevant files. Typically we make directories
for the following types of files:</p>
<ul>
<li>Raw data, called <code>data</code>, <code>input</code>, etc,</li>
<li>Results, often <code>results</code> or <code>output</code> with
subfolders for <code>tables</code>, <code>figures</code>, and
<code>rdata</code>, and</li>
<li>Scripts, often <code>scripts</code>.</li>
</ul>
<p>We’ve already provided the raw data in the <code>inputs/</code>
folder, but you’d otherwise want to move the starting, unaltered, data
into this folder.</p>
<div id="creating-directory-stucture" class="section level2">
<h2>Creating directory stucture</h2>
<p>Before we start creating directories, let’s make sure we’re in the
right location. To print the current working directory:</p>
<pre class="r"><code># =========================================================================
# Day 1 - Getting Started with Functional enrichment
# =========================================================================

# -------------------------------------------------------------------------
# Get current working directory
getwd()</code></pre>
<p>This means that any references to files loaded or files saved
<strong>is with respect to this location</strong>. This can simplify our
code a bit by allowing us to use relative paths rather than full paths.
Let’s set our working directory to the ISC folder in our respective home
directories.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Set current working directory
setwd(&#39;~/IFUN_R&#39;)</code></pre>
<p>Now that we’re sure of our working directory, let’s create some
folders for our analysis scripts and results thereof.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Create directory structure

dir.create(&#39;scripts&#39;, recursive = TRUE, showWarnings = FALSE)
dir.create(&#39;results/figures&#39;, recursive = TRUE, showWarnings = FALSE)
dir.create(&#39;results/tables&#39;, recursive = TRUE, showWarnings = FALSE)
dir.create(&#39;results/rdata&#39;, recursive = TRUE, showWarnings = FALSE)</code></pre>
</div>
<div id="saving-scripts" class="section level2">
<h2>Saving scripts</h2>
<p>Let’s save our currently open script in the <code>scripts/</code>
folder as <code>IFUN_day1.R</code> by clicking <KBD>File</KBD> and then
<KBD>Save</KBD>.</p>
</div>
</div>
<div id="using-webgestaltr" class="section level1">
<h1>Using WebGestaltR</h1>
<p>To start, we’ll load the <code>WebGestaltR</code> library along with
<code>tidyverse</code>, which we’ll use for data processing and
visualization. Both packages were both pre-installed on the workshop
server but later in the workshop we’ll discuss how to use these packages
on other compute resources like GreatLakes.</p>
<pre class="r"><code># =========================================================================
# ORA with WebGestlatR
# =========================================================================

# -------------------------------------------------------------------------
# Load the libraries
library(WebGestaltR)</code></pre>
<pre><code>******************************************</code></pre>
<pre><code>*                                        *</code></pre>
<pre><code>*          Welcome to WebGestaltR !      *</code></pre>
<pre><code>*                                        *</code></pre>
<pre><code>******************************************</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching core tidyverse packages ───────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     </code></pre>
<pre><code>── Conflicts ─────────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<p><em>Note</em> - when loading <code>tidyverse</code> you will likely
see some red text. Remember that this is an informative message about
the package loading and functions that share names between multiple
source packages, not a warning or error. <br></p>
<p>Let’s look at the documentation for the main function in WebGestaltR
to understand how to input these required objects into the function.
Think of it as reading the recipe before you start cooking.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Look at the manual for the main function
?WebGestaltR</code></pre>
<p>There are a lot of arguments that go into this function, but, like
many other R functions, we won’t need or use all of them. Let’s focus on
the arguments we’ll be using, while noting some of the defaults.</p>
<ul>
<li><code>enrichMethod</code> is the method used for the functional
enrichment. We will use <code>'ORA'</code> for this part of the
workshop.</li>
<li><code>organism</code> is the organism used for the experiment. The
selection helps WebGestaltR extract the correct genes for the correct
gene sets.</li>
<li><code>enrichDatabase</code> are the <em>functional/biological
knowledge database</em> with the genesets and pathways that will be
compared to the input data</li>
</ul>
<p>We can use the <code>listOrganism()</code> function to see all
available <code>organisms</code> available for the package, and,
importantly, how they’re labeled. We can expect that the organism ID
will need to be matched exactly in both spelling and case when
specifying that argument.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# List available organisms
listOrganism()</code></pre>
<pre><code> [1] &quot;athaliana&quot;     &quot;btaurus&quot;       &quot;celegans&quot;      &quot;cfamiliaris&quot;   &quot;drerio&quot;       
 [6] &quot;sscrofa&quot;       &quot;dmelanogaster&quot; &quot;ggallus&quot;       &quot;hsapiens&quot;      &quot;mmusculus&quot;    
[11] &quot;rnorvegicus&quot;   &quot;scerevisiae&quot;  </code></pre>
<p><br></p>
<p>To see what functional databases are available from the tool, we can
see the <code>listGeneSet()</code> function to see the available
genesets for specific organisms. For this function, we need to specify
<code>mmusculus</code> as the organism to have the appropriate
references listed.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# List available reference genesets for mmusculus
listGeneSet(organism = &#39;mmusculus&#39;) %&gt;% head(10)</code></pre>
<pre><code>                                          name
1              geneontology_Biological_Process
2  geneontology_Biological_Process_noRedundant
3              geneontology_Cellular_Component
4  geneontology_Cellular_Component_noRedundant
5              geneontology_Molecular_Function
6  geneontology_Molecular_Function_noRedundant
7                                 pathway_KEGG
8                              pathway_Panther
9                             pathway_Reactome
10                         pathway_Wikipathway
                                                                                                                                                                                                                                                                                                 description
1                                                                                                                                                                                                            The gene ontology biological process database was downloaded from http://www.geneontology.org/.
2  The gene ontology biological process database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.
3                                                                                                                                                                                                            The gene ontology cellular component database was downloaded from http://www.geneontology.org/.
4  The gene ontology cellular component database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.
5                                                                                                                                                                                                            The gene ontology molecular function database was downloaded from http://www.geneontology.org/.
6  The gene ontology molecular function database was downloaded from http://www.geneontology.org/. Then, we only contain the non-redundant categories by selecting the most general categories in each branch of the GO DAG structure from all categories with the number of annotated genes from 20 to 500.
7                                                                                                                                                                                                                                         The KEGG pathway database was downloaded from http://www.kegg.jp/.
8                                                                                                                                                                                                                        The PANTHER pathway database was downloaded from http://www.pantherdb.org/pathway/.
9                                                                                                                                                                                                                                The Reactome pathway database was downloaded from http://www.reactome.org/.
10                                                                                                                                                                                                                                 The Wikipathway database was downloaded from http://www.wikipathway.org/.
       idType
1  entrezgene
2  entrezgene
3  entrezgene
4  entrezgene
5  entrezgene
6  entrezgene
7  entrezgene
8  entrezgene
9  entrezgene
10 entrezgene</code></pre>
<p>Here we should see some familar options, including
<code>geneontology_Biological_Process_noRedundant</code>.</p>
<p>Continuing with the arguments for the <code>WebGestaltR()</code>
function:</p>
<ul>
<li><code>interestGene</code> for ORA this should be a vector of gene
identifiers for a set of differentially expressed “interest” genes.</li>
<li><code>interestGeneType</code> is the type of identifier used to
describe the genes.</li>
</ul>
<p>We can use the <code>listIdType()</code> function to see which
identifiers are available for which organisms, adding on a
<code>tail()</code> to look at a subset. We will be using the
<code>'genesymbol'</code> type, but note the presence of
<code>'ensemble_gene_id'</code> and <code>'entrezgene'</code>, among
other common (and not-so-common) identifiers.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# List gene identifier types for mmusculus
listIdType(organism = &#39;mmusculus&#39;) %&gt;% tail(25)</code></pre>
<pre><code> [1] &quot;dbSNP&quot;                       &quot;embl&quot;                       
 [3] &quot;ensembl_gene_id&quot;             &quot;ensembl_peptide_id&quot;         
 [5] &quot;entrezgene&quot;                  &quot;entrezgene_protein-coding&quot;  
 [7] &quot;genename&quot;                    &quot;genesymbol&quot;                 
 [9] &quot;illumina_ScriptSeq_RNASeqV2&quot; &quot;illumina_WG6_Customarray&quot;   
[11] &quot;illumina_mouseref_8&quot;         &quot;illumina_mouseref_8_v2&quot;     
[13] &quot;illumina_mousewg_6_v1&quot;       &quot;illumina_mousewg_6_v2&quot;      
[15] &quot;interpro&quot;                    &quot;phalanx_onearray&quot;           
[17] &quot;phosphositeEnsembl&quot;          &quot;phosphositeRefseq&quot;          
[19] &quot;phosphositeSeq&quot;              &quot;phosphositeUniprot&quot;         
[21] &quot;protein_id&quot;                  &quot;refseq_mrna&quot;                
[23] &quot;refseq_peptide&quot;              &quot;unigene&quot;                    
[25] &quot;uniprotswissprot&quot;           </code></pre>
<p>Continuing with the arguments for the function:</p>
<ul>
<li><code>referenceGene</code> is also a vector of gene identifiers, but
this is the background set of genes. Unlike the browser option, here
will use all the genes tested for differential expression in the DE
analysis, noting that this itself implies some prior filtering.</li>
<li><code>referenceGeneType</code> is again the type of identifier used,
but to describe the reference set.</li>
<li><code>fdrThr</code> refers to the significance threshold for the
<code>fdr</code> method (by default the Benjamini-Hochberg FDR
correction). The default is <code>fdrThre = 0.05</code>, but you may
want to relax this threshold to see what genesets are “close” to
significance, depending on the limitations of the DE results and
downstream goals.</li>
<li><code>outputDirectory</code> and <code>projectName</code> are used
to specify where to put results and what to call them. By default,
<code>WebGestaltR()</code> has rich output in the form of supporting
files, but also an HTML report (just like the web-version) that
summarizes the results and provides helpful visualizations.</li>
</ul>
<p>We can think of the above arguments as the minimum necessary for an
over-representation analysis. Now let’s read in the differential
expression results, and create the right inputs for the ORA.</p>
<div id="read-in-diffex-results" class="section level2">
<h2>Read in diffex results</h2>
<p>For our first ORA, we will use the differential expression results
from the same bulk RNA-seq data we used in the browser-based analysis
from the RNA-seq Demystified workshop <a
href="https://umich-brcf-bioinf.github.io/workshop-rnaseq-demystified/main/html/Module07_DESeq2Init.html#Sample_Information">(link)</a>.</p>
<p>We have provided these differential results as a CSV file, which we
can read in with the <code>read_csv()</code> function:</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Read diffex results
rsd_diffex = read_csv(&#39;inputs/bulk_de_results/de_deficient_vs_control_annotated.csv&#39;)</code></pre>
<pre><code>Rows: 16249 Columns: 9
── Column specification ─────────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (3): id, symbol, call
dbl (6): baseMean, log2FoldChange, lfcSE, stat, pvalue, padj

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<p>As always, it’s helpful to look at the result of what we read in to
get a sense for how to work with it:</p>
<pre class="r"><code># -------------------------------------------------------------------------
head(rsd_diffex)</code></pre>
<pre><code># A tibble: 6 × 9
  id                 symbol baseMean log2FoldChange lfcSE   stat pvalue   padj call 
  &lt;chr&gt;              &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;
1 ENSMUSG00000000001 Gnai3   1490.            0.278 0.148  1.88  0.0605  0.325 NS   
2 ENSMUSG00000000028 Cdc45   1749.            0.222 0.129  1.72  0.0853  0.386 NS   
3 ENSMUSG00000000031 H19     2152.            0.136 0.284  0.478 0.633   0.867 NS   
4 ENSMUSG00000000037 Scml2     24.9           0.600 0.562  1.07  0.286  NA     NS   
5 ENSMUSG00000000049 Apoh       7.78         -1.23  1.15  -1.07  0.285  NA     NS   
6 ENSMUSG00000000056 Narf   19654.           -0.201 0.167 -1.20  0.229   0.596 NS   </code></pre>
</div>
<div id="extracting-genes-for-ora" class="section level2">
<h2>Extracting genes for ORA</h2>
<p>We can see some of the typical output from DESeq2, including a gene
identifier (or two), overall expression, log fold-change, p-value, and
adjusted p-value (FDR). As we saw in the documentation for
<code>WebGestaltR()</code>, we’ll need two lists of gene
identifiers:</p>
<ol style="list-style-type: decimal">
<li><code>interestGene</code>: The gene identifiers representing the DE
genes, and</li>
<li><code>referenceGene</code>: The gene identifiers representing a
background set of genes, ideally all genes tested in the DE
analysis.</li>
</ol>
<p>For simplicity, and slightly easier interpretability, let’s use the
gene symbol column. It’s always a good idea to see how many identifiers
are NA, and remove them because they can’t be used in the analysis
anyhow.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# How many symbols are NA?
table(is.na(rsd_diffex$symbol))</code></pre>
<pre><code>
FALSE  TRUE 
16114   135 </code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Filter out the NAs
rsd_diffex = rsd_diffex %&gt;% dplyr::filter(!is.na(symbol))

# Verify we have fewer by noting dimension of resulting table
rsd_diffex</code></pre>
<pre><code># A tibble: 16,114 × 9
   id                 symbol baseMean log2FoldChange lfcSE   stat pvalue   padj call 
   &lt;chr&gt;              &lt;chr&gt;     &lt;dbl&gt;          &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;
 1 ENSMUSG00000000001 Gnai3   1490.            0.278 0.148  1.88  0.0605  0.325 NS   
 2 ENSMUSG00000000028 Cdc45   1749.            0.222 0.129  1.72  0.0853  0.386 NS   
 3 ENSMUSG00000000031 H19     2152.            0.136 0.284  0.478 0.633   0.867 NS   
 4 ENSMUSG00000000037 Scml2     24.9           0.600 0.562  1.07  0.286  NA     NS   
 5 ENSMUSG00000000049 Apoh       7.78         -1.23  1.15  -1.07  0.285  NA     NS   
 6 ENSMUSG00000000056 Narf   19654.           -0.201 0.167 -1.20  0.229   0.596 NS   
 7 ENSMUSG00000000078 Klf6     951.           -0.131 0.166 -0.788 0.430   0.752 NS   
 8 ENSMUSG00000000085 Scmh1    727.            0.232 0.186  1.25  0.212   0.577 NS   
 9 ENSMUSG00000000088 Cox5a   1427.           -0.264 0.205 -1.29  0.198   0.556 NS   
10 ENSMUSG00000000093 Tbx2      15.2           0.548 0.663  0.826 0.409  NA     NS   
# ℹ 16,104 more rows</code></pre>
<p>Let’s start with the reference set because it doesn’t require any
filtering:</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Pull all genes tested
rsd_ref_ora = rsd_diffex %&gt;% pull(symbol)

# -------------------------------------------------------------------------
# Preview the vector
head(rsd_ref_ora)</code></pre>
<pre><code>[1] &quot;Gnai3&quot; &quot;Cdc45&quot; &quot;H19&quot;   &quot;Scml2&quot; &quot;Apoh&quot;  &quot;Narf&quot; </code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# How big is the custom background?
length(rsd_ref_ora)</code></pre>
<pre><code>[1] 16114</code></pre>
<p>Next, let’s extract the significantly differentially expressed genes.
We’ll use the same thresholds used to call DE genes in the original
analysis - a “typical” significance threshold of FDR &lt; 0.05 and
|log2FC| &gt; log2(1.5).</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Pull diffex genes
rsd_sig_ora = rsd_diffex %&gt;%
    dplyr::filter(padj &lt; 0.05 &amp; abs(log2FoldChange) &gt; log2(1.5)) %&gt;%
    pull(symbol)

# -------------------------------------------------------------------------
# Preview the vector
head(rsd_sig_ora)</code></pre>
<pre><code>[1] &quot;Trim25&quot; &quot;Bcl11a&quot; &quot;Itgb7&quot;  &quot;Ddb2&quot;   &quot;Apoe&quot;   &quot;Gys1&quot;  </code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# How many significant genes are there?
length(rsd_sig_ora)</code></pre>
<pre><code>[1] 189</code></pre>
</div>
<div id="write-out-gene-lists-to-file" class="section level2">
<h2>Write out gene lists to file</h2>
<p>We can write out the list of DE interest genes to file re-create the
input file we uploaded for the browser-based WebGestalt analysis as well
as output to file the custom reference/background gene list.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Write out list of DE interest genes to file
write_lines(rsd_sig_ora, file=&quot;./results/tables/deficient_DE_GeneList.txt&quot;)

# -------------------------------------------------------------------------
# Write out custom reference/background set of genes to file
write_lines(rsd_ref_ora, file=&quot;./results/tables/deficient_reference_GeneList.txt&quot;)</code></pre>
<p>To download these tables, we can use the file viewer tab to navigate
to the <code>tables</code> subdirectory within our <code>results</code>
directory, select the two files, and then use the blue <KBD>Gear</KBD>
symbol on the top right of the pane to select <code>Export</code>. After
re-naming the export <code>tables-export.zip</code> and selecting
<KBD>Download</KBD>, the files should be in a zipped folder in the
downloads folder on your computer.</p>
<p><img src="images/content/03-WebGestalt_R/08_restudio_download_tables.png" alt="rstudio download file" style="width:1000px;"/></p>
<p><br></p>
</div>
</div>
<div id="running-ora" class="section level1">
<h1>Running ORA</h1>
<p>We now have an understanding of the important parameters and vectors
of genes prepared to follow through with the over-representation
analysis. For demonstrative purposes we will start with the same
non-redundant version of the Gene Ontology Biological Process genesets
used in the web version but relax our significance threshold.
<!-- repeat for 'pathway_KEGG' --></p>
<pre class="r"><code># -------------------------------------------------------------------------
# ORA on bulk RNA-seq DESeq2 results with GO - Biological Process knowledge base
rsd_ora_result = WebGestaltR(
    enrichMethod = &#39;ORA&#39;,
    organism = &#39;mmusculus&#39;,
    enrichDatabase = c(&#39;geneontology_Biological_Process_noRedundant&#39;),
    interestGene = rsd_sig_ora,
    referenceGene = rsd_ref_ora,
    interestGeneType = &#39;genesymbol&#39;,
    referenceGeneType = &#39;genesymbol&#39;,
    fdrThr = 0.1,
    outputDirectory = &#39;./results&#39;,
    projectName = &#39;deficient_vs_control_ORA-GO_BP&#39;)</code></pre>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...</code></pre>
<pre><code>Warning in dir.create(projectDir):
&#39;./results/Project_deficient_vs_control_ORA_GO_BP&#39; already exists</code></pre>
<pre><code>Summarizing the input ID list by GO Slim data...</code></pre>
<pre><code>Performing the enrichment analysis...
Begin affinity propagation...
End affinity propagation...
Begin weighted set cover...
Remain is 0, ending weighted set cover
Generate the final report...
Results can be found in the ./results/Project_deficient_vs_control_ORA_GO_BP!</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# View the first few results
head(rsd_ora_result)</code></pre>
<pre><code>     geneSet                                                     description
1 GO:0070997                                                    neuron death
2 GO:0098754                                                  detoxification
3 GO:0062012                  regulation of small molecule metabolic process
4 GO:0006575                  cellular modified amino acid metabolic process
5 GO:0097193                           intrinsic apoptotic signaling pathway
6 GO:0043620 regulation of DNA-templated transcription in response to stress
                                                 link size overlap    expect
1 http://amigo.geneontology.org/amigo/term/GO:0070997  263      14 3.7281697
2 http://amigo.geneontology.org/amigo/term/GO:0098754   77       7 1.0915174
3 http://amigo.geneontology.org/amigo/term/GO:0062012  235      12 3.3312543
4 http://amigo.geneontology.org/amigo/term/GO:0006575  120       8 1.7010660
5 http://amigo.geneontology.org/amigo/term/GO:0097193  227      11 3.2178499
6 http://amigo.geneontology.org/amigo/term/GO:0043620   50       5 0.7087775
  enrichmentRatio       pValue        FDR
1        3.755194 1.979515e-05 0.01565796
2        6.413091 1.026862e-04 0.03144459
3        3.602247 1.192589e-04 0.03144459
4        4.702933 2.905229e-04 0.05745090
5        3.418432 3.635356e-04 0.05751133
6        7.054400 6.718740e-04 0.08857539
                                                                             overlapId
1 11816;12048;12176;12608;12721;13198;14381;15258;16476;17748;22339;26570;80890;112405
2                                       11816;14782;17748;17750;76650;230784;100503605
3          11816;12048;12724;14783;26570;59090;67300;77781;227358;228775;230784;266645
4                                    14381;14782;17768;18451;26570;69065;107869;112405
5                  12048;12176;12575;12608;13198;15258;56312;59041;69065;228775;242705
6                                                       12608;13198;16476;22339;230784
                                                                              userId
1 Apoe;Bcl2l1;Vegfa;Ddit3;Slc7a11;Trim2;Coro1a;G6pdx;Mt1;Egln1;Jun;Cebpb;Hipk2;Bnip3
2                                                Apoe;Sesn2;Gsr;Mt2;Mt1;Srxn1;Hbb-bs
3          Apoe;Bcl2l1;Grb10;Clcn2;Acmsd;Slc7a11;Sesn2;Trib3;Midn;Epm2aip1;Cltc;Erfe
4                                     Mthfd2;P4ha1;Chac1;Slc7a11;Cth;G6pdx;Gsr;Egln1
5                 Bcl2l1;E2f2;Cdkn1a;Ddit3;Stk25;Chac1;Nupr1;Trib3;Cebpb;Hipk2;Bnip3
6                                                        Vegfa;Ddit3;Sesn2;Jun;Cebpb</code></pre>
<p>The columns of the table are similar to the results displayed in the
web-version with some additional details:</p>
<ul>
<li><code>geneSet</code>: The database-specific identifier for the
geneset/pathway.</li>
<li><code>description</code>: A readable description of the
geneset/pathway.</li>
<li><code>link</code>: A link to the corresponding database giving more
detail about the geneset/pathway.</li>
<li><code>size</code>: The number of genes in the geneset.</li>
<li><code>overlap</code>: The number of significantly differentially
expressed genes from your dataset overlapping the geneset.</li>
<li><code>expect</code>: The expected number of input genes annotated to
the geneset.</li>
<li><code>enrichmentRatio</code>: The actual / expected overlap.</li>
<li><code>pValue</code>: the p-value from the hypergeometric test.</li>
<li><code>FDR</code>: The corrected p-value using the Benjamini-Hochberg
method.</li>
<li><code>overlapID</code>: The Entrez Gene IDs overlapping the
geneset.</li>
<li><code>database</code>: The database of the geneset.</li>
<li><code>userId</code>: The user-supplied gene ID overlapping the
geneset.</li>
</ul>
<p>To expand our analysis and aid with interpretation of our DE results,
we’ll run a second enrichment adding a different knowledge base called
<a href="https://www.genome.jp/kegg/">KEGG (Kyoto Encyclopedia of Genes
and Genomes)</a> that per the website “is a database resource for
understanding high-level functions and utilities of the biological
system”. Broadly speaking, KEGG includes gene sets for known pathways as
well as other functional annotations. Here we’ll also diverge from our
previous enrichment by reducing the maximum gene number per category to
limit the comparison to smaller, more specific categories.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# ORA on bulk RNA-seq DESeq2 results with KEGG knowledge base
rsd_ora_kegg_result = WebGestaltR(
    enrichMethod = &#39;ORA&#39;,
    organism = &#39;mmusculus&#39;,
    enrichDatabase = c(&#39;pathway_KEGG&#39;),
    interestGene = rsd_sig_ora,
    referenceGene = rsd_ref_ora,
    interestGeneType = &#39;genesymbol&#39;,
    referenceGeneType = &#39;genesymbol&#39;,
    fdrThr = 0.1,
    maxNum = 60,
    outputDirectory = &#39;./results&#39;,
    projectName = &#39;deficient_vs_control_ORA-KEGG&#39;)</code></pre>
<pre><code>Loading the functional categories...
Loading the ID list...
Loading the reference list...</code></pre>
<pre><code>Summarizing the input ID list by GO Slim data...</code></pre>
<pre><code>Performing the enrichment analysis...
Begin affinity propagation...
End affinity propagation...
Begin weighted set cover...
Remain is 0, ending weighted set cover
Generate the final report...
Results can be found in the ./results/Project_deficient_vs_control_ORA_KEGG!</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# View the first few results
head(rsd_ora_kegg_result)</code></pre>
<pre><code>   geneSet                 description
1 mmu01230 Biosynthesis of amino acids
2 mmu04978          Mineral absorption
                                                                                      link
1 http://www.kegg.jp/kegg-bin/show_pathway?mmu01230+100678+107869+209027+27053+56421+56454
2                http://www.kegg.jp/kegg-bin/show_pathway?mmu04978+12724+14325+17748+17750
  size overlap    expect enrichmentRatio       pValue        FDR
1   60       6 0.9919513        6.048684 0.0004125129 0.07260228
2   28       4 0.4629106        8.640977 0.0010455220 0.09200593
                               overlapId                            userId
1 100678;107869;209027;27053;56421;56454 Pfkp;Aldh18a1;Pycr1;Cth;Psph;Asns
2                12724;14325;17748;17750                Clcn2;Mt2;Mt1;Ftl1</code></pre>
<p>Although we limited the comparisons to more specific terms, we only
find two pathways that have significant enrichment below an FDR &lt;
0.10 threshold. This could be because the treatment doesn’t impact a
specific pathway or it could be due to the limits of ORA, which requires
we subset our measured genes into DE and non-DE and also doesn’t utilize
the effect size data (e.g. fold-change).</p>
<!-- TODO: Here is where we'll actually talk more about the biological relevance of the results and pat ourselves on the back for our first ORA well done. -->
</div>
<div id="results-of-ora" class="section level1">
<h1>Results of ORA</h1>
<p>The R package for WebGestalt outputs a number of useful artifacts to
an <code>outputDirectory</code> and <code>projectName</code> of our
choosing. Some of the more important output files are:</p>
<ul>
<li><code>enrichment_results_{project name}.txt</code>: A tab-delimited
text table of the full results which can be read in Excel. It is the
same as the <code>rsd_ora_result</code> table.</li>
<li><code>interestingID_mappingTable_{project name}.txt</code>: A table
of the <code>interestGene</code> vector, with user-supplied identifiers
and mappings to other identifiers.</li>
<li><code>interestingID_unmappedList_{project_name}.txt</code>: The
identifiers which could not be mapped. <strong>This may be helpful to
know if any key genes were dropped from the analysis.</strong></li>
<li><code>Report_{project_name}.html</code>: An HTML report, like the
web application, of the results. This may be more suitable for a first
glance and if we review it, it looks quite familiar.</li>
</ul>
<!-- enrichment_results_deficient_vs_control_ORA_GO_BP = file with terms and list of genes (in column "userId", split by semi-columns -->
</div>
<div id="clean-up-session" class="section level1">
<h1>Clean up session</h1>
<pre class="r"><code># -------------------------------------------------------------------------
# Clean up session to enable quick spin up tomorrow
clear_list = c(&quot;rsd_ora_kegg_result&quot;, 
          &quot;rsd_ora_result&quot;, 
          &quot;rsd_ref_ora&quot;, 
          &quot;rsd_sig_ora&quot;)

# Only remove plots that actually exist in the environment
rm(list=Filter(exists, clear_list))
gc()</code></pre>
<pre><code>          used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
Ncells 2789138 149.0    4950962 264.5         NA  4950962 264.5
Vcells 5807211  44.4   12263262  93.6      36864 12263262  93.6</code></pre>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>In this section, we performed our first ORA with the WebGestalt R
package. In so doing, we accomplished the following objectives:</p>
<ul>
<li>Form the inputs for over-representation analysis (ORA) in
WebGestaltR from a differential expression analysis.</li>
<li>Use WebGestaltR to do an ORA analysis.</li>
<li>Understand how to interpret the results of ORA analysis.</li>
</ul>
</div>
<div id="questions" class="section level1">
<h1>Questions</h1>
<ul>
<li>What are the advantages of running WebGestaltR over using the
browser version?</li>
<li>What differences in the results might you expect if an identifier
other than the gene symbol was used?</li>
</ul>
<br/> <br/>
<hr/>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"><a href="02-IntroToWebGestaltandORA.html">Previous
lesson</a></th>
<th align="center"><a href="#top">Top of this lesson</a></th>
<th align="right"><a href="07A-advanced-visualizations.html">Next
lesson</a></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
