<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GSEA with R</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link rel="shortcut icon" href="favicon-16x16.png" />
<!--
Favicon from built via
https://favicon.io/emoji-favicons/computer/
-->
<meta name="google-site-verification" content="t-Wgs-RXt8xi94RSoFamxoD2P2h-Hh7M_t_iOuxFZUo" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intro Functional Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="workshop-intro.html">Intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-overview.pdf">Functional analysis and ORA</a>
    </li>
    <li>
      <a href="02-IntroToWebGestaltandORA.html">ORA with WebGestalt</a>
    </li>
    <li>
      <a href="03-WebGestaltRORA.html">ORA with R</a>
    </li>
    <li>
      <a href="07A-advanced-visualizations.html">Advanced visualizations I</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-gsea-overview.pdf">Functional Analysis and GSEA</a>
    </li>
    <li>
      <a href="06-WebGestaltRGSEA.html">GSEA with R</a>
    </li>
    <li>
      <a href="gene-set-references.pdf">GO, KEGG, and friends</a>
    </li>
    <li>
      <a href="07B-advanced-visualizations.html">Advanced visualizations II</a>
    </li>
    <li>
      <a href="08-analysis-summary.html">Analysis summary and next steps</a>
    </li>
  </ul>
</li>
<li>
  <a href="workshop-wrap-up.html">Wrap up</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GSEA with R</h1>

</div>


<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>
<script>
  addClassKlippyTo("pre.r, pre.markdown, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<p><img src="images/wayfinder/wayfinder-06A.png"
alt="workshop wayfinder" /></p>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<ul>
<li>Form the inputs for Gene Set Enrichment Analysis (GSEA) in
WebGestaltR from single-cell results.</li>
<li>Use WebGestaltR to do a GSEA analysis.</li>
<li>Understand how to interpret the results of GSEA analysis.</li>
<li>Compare and contrast the results of two GSEA results.</li>
</ul>
</div>
<div id="getting-started" class="section level1">
<h1>Getting started</h1>
<p>The setup for GSEA is similar to ORA, but there are two key
differences:</p>
<ol style="list-style-type: decimal">
<li>We do not require a threshold for significance, and</li>
<li>We need to pass gene identifiers <strong>and</strong> the log
fold-change for the gene from the differential expression test.</li>
</ol>
<p>This gives the advantage of being able to use the differential
expression results more fully, instead of limiting to DE genes.</p>
</div>
<div id="running-gsea-using-webgestalt" class="section level1">
<h1>Running GSEA using WebGestalt</h1>
<p>First, we’ll re-load the libraries to ensure they are available.</p>
<pre class="r"><code># =========================================================================
# ORA with WebGestlatR
# =========================================================================

# -------------------------------------------------------------------------
# Load the libraries
library(WebGestaltR)
library(tidyverse)</code></pre>
<p>We’ll then will take a second look at the documentation for the
<code>WebGestaltR()</code> function, and note the difference in input
between ORA and GSEA.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Look at the manual for the main function
?WebGestaltR</code></pre>
<p>We note from the documentation that <code>interestGene</code> “should
be an R data.frame object containing two columns: the gene list and the
corresponding scores.” We also note that we will <strong>not</strong>
make use of the <code>referenceGene</code> argument.</p>
<div id="read-in-diffex-results" class="section level2">
<h2>Read in diffex results</h2>
<p>We will demonstrate GSEA on a differential analysis from the
single-cell RNA-seq experiment we explored in the Introduction to
Single-cell Analysis <a
href="https://umich-brcf-bioinf.github.io/workshop-intro-single-cell/main/html/00A-OrientingOnScRNASeq.html#Consider_a_specific_scRNA-Seq_experiment">(link)</a>.
In particular, we’ll perform GSEA on the differential expression results
in the pericyte cluster, starting with the results from the day 21 to
day 7 comparison.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D21 vs D7 in pericyte cluster
isc_d21_v_d7 = read_csv(&#39;inputs/single_cell_de_results/de_pseudo_pericyte_D21_vs_D7.csv&#39;)</code></pre>
<pre><code>Rows: 23608 Columns: 6
── Column specification ───────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): gene
dbl (5): p_val, avg_log2FC, pct.1, pct.2, p_val_adj

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Preview the result
head(isc_d21_v_d7)</code></pre>
<pre><code># A tibble: 6 × 6
      p_val avg_log2FC pct.1 pct.2 p_val_adj gene   
      &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;  
1 3.86e-232       2.91     1     1 1.02e-227 Cilp2  
2 5.63e-118       1.71     1     1 1.49e-113 Ltbp4  
3 1.85e- 95       1.29     1     1 4.91e- 91 Cd55   
4 2.60e- 87       1.64     1     1 6.88e- 83 Prss23 
5 3.02e- 87      -1.59     1     1 8.00e- 83 Nav3   
6 7.46e- 82       2.25     1     1 1.98e- 77 Gm42418</code></pre>
<p>Before preparing our data for GSEA, let’s also check how many genes
were DE in the pseudobulk D21 vs D7 comparison using the same thresholds
from the workshop.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D21 vs D7 in pericyte cluster
table(isc_d21_v_d7$p_val_adj &lt; 0.05 &amp; abs(isc_d21_v_d7$avg_log2FC) &gt; 1.5)</code></pre>
<pre><code>
FALSE  TRUE 
23553    55 </code></pre>
<!-- NOTE - mistake in ISC results - miss log2 transformation of threshold in table step (https://umich-brcf-bioinf.github.io/workshop-intro-single-cell/main/html/08-DifferentialExpression.html#Pseudobulk_comparisons) so the number of DE genes artificially  deflated and actually # of DE genes for scRNA is ~400 so is likely a better candidate for the ORA over the bulk results -->
</div>
<div id="extracting-data-for-gsea" class="section level2">
<h2>Extracting data for GSEA</h2>
<p>The key columns to extract from our Day 21 vs Day 7 pericyte results
table, as per the documentation, are the <code>gene</code> and a column
of values to use to rank the genes. Before proceeding, it’s prudent to
check if there are missing values in the column we’ll use as our gene
identifier:</p>
<pre class="r"><code># -------------------------------------------------------------------------
# How many symbols are NA?
table(is.na(isc_d21_v_d7$gene))</code></pre>
<pre><code>
FALSE 
23608 </code></pre>
<p>For ranking, we can use the values from the <code>avg_log2FC</code>
column and create a table with just the <code>gene</code> and
<code>avg_log2FC</code> columns with the <code>select()</code> function
from <code>dplyr</code>:</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Select gene and avg_log2FC columns
isc_d21_v_d7_gsea = isc_d21_v_d7 %&gt;%
    dplyr::select(gene, avg_log2FC)

# -------------------------------------------------------------------------
# Preview the table
head(isc_d21_v_d7_gsea)</code></pre>
<pre><code># A tibble: 6 × 2
  gene    avg_log2FC
  &lt;chr&gt;        &lt;dbl&gt;
1 Cilp2         2.91
2 Ltbp4         1.71
3 Cd55          1.29
4 Prss23        1.64
5 Nav3         -1.59
6 Gm42418       2.25</code></pre>
</div>
</div>
<div id="running-gsea" class="section level1">
<h1>Running GSEA</h1>
<p>We will run GSEA on our single-cell results using the <a
href="https://www.genome.jp/kegg/pathway.html">KEGG pathway knowledge
base</a>, which is per the website “a collection of manually drawn
pathway maps representing our knowledge of the molecular interaction,
reaction and relation networks” instead of gene ontology. This option
allows us to</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Run GSEA
isc_d21_v_d7_gsea_result = WebGestaltR(
    enrichMethod = &#39;GSEA&#39;,
    nThreads = 8,
    organism = &#39;mmusculus&#39;,
    enrichDatabase = c(&#39;pathway_KEGG&#39;),
    interestGene = isc_d21_v_d7_gsea,
    interestGeneType = &#39;genesymbol&#39;,
    fdrThr = 0.1,
    outputDirectory = &#39;./results&#39;,
    projectName = &#39;D21_v_D7_GSEA_KEGG&#39;, 
    cache = NULL)</code></pre>
<pre><code>Loading the functional categories...
Loading the ID list...</code></pre>
<pre><code>Summarizing the uploaded ID list by GO Slim data...</code></pre>
<pre><code>Performing the enrichment analysis...</code></pre>
<pre><code>1000 permutations of score complete...</code></pre>
<pre><code>Begin affinity propagation...
End affinity propagation...
Begin weighted set cover...
End weighted set cover...
Generate the final report...
Results can be found in the ./results/Project_D21_v_D7_GSEA_KEGG!</code></pre>
<pre class="r"><code># clean up session
gc()</code></pre>
<pre><code>          used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells 2801880 149.7    5204824 278.0   5204824 278.0
Vcells 5800826  44.3  103859791 792.4 129823836 990.5</code></pre>
<p><br></p>
<p>One downside of GSEA is that it is more computationally demanding
than ORA. As the enrichment runs, we’ll see the reporting messages
<code>1000 permutations of score complete</code> this permutation was
introduced in the overview and is what generates a null distribution to
be able to determine “how surprising” the ranked overlaps are.</p>
<p>But what about our input? Why did we include the fold-change
column?</p>
<p>As mentioned, a ranking metric is needed to order the genes before
they are compared to the functional gene sets to generate the enrichment
score.</p>
<p>By using log fold-changes, we’re using the effect size of difference
between the experimental groups, the equivalent of
<code>log2_Ratio_of_Classes</code> in <a
href="https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideFrame.html">Broad’s
GSEA user guide</a> describes the several ranking metrics available for
that tool. The advantage of using this as our ranking metric is that we
are using the biologically relevant difference while ignoring
significance level, which is important to rule out false positive but
can be impacted by within group variation and limited samples which
reduce the power to detect true positives.</p>
<p>However, you may also see a local ranking statistic that combines
both the direction of the effect size and the significance
(e.g. <code>-log10({p value}) * sign({Fold Change})</code>). The
advantage of this approach is that it incorporates the significance
level to protect against false positives - unlike fold-change alone -
while still using the direction of the effect size. The disadvantage of
using this combined metric is that the <em>magnitude</em> of the effect
size observed in the experiment is no longer part of the ranking.</p>
<p><img
src="images/content/06-WebGestalt_GSEA/PathwayCommons_figure_gsea_global.png" /></p>
<!-- what else to add here?-->
<p>For more context, this <a
href="https://www.pathwaycommons.org/guide/primers/data_analysis/gsea/">Pathways
Common guide</a> delves into the history and describes the GSEA
algorithm in more detail.</p>
<p>The original version of GSEA is described in <a
href="https://pubmed.ncbi.nlm.nih.gov/12808457/">Mootha 2003</a> the
values were chosen specifically such that the sum over all genes would
be zero. An updated procedure using a ‘weighted’ version of such that
the increments to the running sum are proportional to the rank metric
for that gene is described in <a
href="https://www.pnas.org/doi/10.1073/pnas.0506580102">Subramanian et
al 2005</a> and is the GSEA method used by <a
href="https://cran.r-project.org/web/packages/WebGestaltR/WebGestaltR.pdf">WebGestaltR
based on the package’s reference manual</a>.</p>
</div>
<div id="results-of-gsea" class="section level1">
<h1>Results of GSEA</h1>
<p>Once the tool is done running, we can review the results. As with the
ORA results, the GSEA method outputs similar results tables, gene
identifier mappings, and HTML reports to file but includes a table of
the results in session.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# View the first few results
head(isc_d21_v_d7_gsea_result)</code></pre>
<pre><code>   geneSet                      description
1 mmu04512         ECM-receptor interaction
2 mmu03050                       Proteasome
3 mmu03030                  DNA replication
4 mmu03040                      Spliceosome
5 mmu04657          IL-17 signaling pathway
6 mmu04974 Protein digestion and absorption
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  link
1                                                                                                                                                                                                                                                                                                                                    http://www.kegg.jp/kegg-bin/show_pathway?mmu04512+12824+12845+12643+12841+21825+12840+21828+12839+12827+16779+20971+226519+12826+213119+12843+13138+16773+15530+12842+21827+15891+12834+16404+12833+11603+21826+319480+16398+12835+16776+81877+16419+14268+109700
2                                                                                                                                                                                                                                                                                                                     http://www.kegg.jp/kegg-bin/show_pathway?mmu03050+19192+56436+16912+26440+66997+70247+19177+22123+20422+23996+19166+19179+19167+19184+19173+17463+19186+69077+26443+66413+23997+21762+19172+66537+103554+19171+26442+19181+19170+19188+19185+26445+26441+26446+26444+19175+16913
3                                                                                                                                                                                                                                                                                                                                                                    http://www.kegg.jp/kegg-bin/show_pathway?mmu03030+19076+18968+18974+106344+68275+18972+19075+69724+68209+19687+18971+59001+72151+69263+17217+381760+67153+17220+17215+14156+19718+17218+68240+19891+17216+17219+18538+16881+66979
4 http://www.kegg.jp/kegg-bin/show_pathway?mmu03040+57905+20637+54633+66492+68879+20639+331401+72654+53317+320632+212880+66810+67039+68183+56031+72047+66354+56194+20624+60365+20222+22185+76522+81898+101943+66441+20643+230596+68092+13204+67465+66055+76938+68479+229279+20826+20638+68988+17149+56009+68011+27756+20630+75062+69878+107701+101214+56215+319322+66585+20382+19655+23983+73666+231889+107686+67959+67996+27366+51810+108014+20641+78651+66094+15381+108121+28000+68981+67332+15387+66101+53607+53817+76936+19656+20462+67678+68816+66125+66373+110809+50783+20383+15382+225027+21681
5                                                                                                                                                                                                                                                                                                                                                              http://www.kegg.jp/kegg-bin/show_pathway?mmu04657+14283+171095+17392+12370+71609+15519+15945+22033+14282+21929+16819+19225+22032+15568+18033+22027+16476+12367+110809+18035+12608+17395+20296+16176+20306+14825+20202+20201+20310+20311
6                                                                                                                                                                                                                                                                                                                                                                                             http://www.kegg.jp/kegg-bin/show_pathway?mmu04974+12824+12841+12815+12840+12839+12819+12827+12814+12826+12843+12842+13717+12834+11931+20514+12833+11933+73626+12822+12835+20540+11936+12816+69700+245026
  enrichmentScore normalizedEnrichmentScore pValue          FDR size
1       0.7715357                  2.073585      0 0.0000000000   82
2      -0.8164866                 -1.913958      0 0.0000000000   46
3      -0.8525372                 -1.921080      0 0.0000000000   35
4      -0.7258237                 -1.971289      0 0.0000000000  128
5      -0.6895711                 -1.777504      0 0.0004867660   86
6       0.7163359                  1.928244      0 0.0005231424   82
                                        plotPath leadingEdgeNum
1 ./Project_D21_v_D7_GSEA_KEGG_GSEA/mmu04512.png             34
2 ./Project_D21_v_D7_GSEA_KEGG_GSEA/mmu03050.png             37
3 ./Project_D21_v_D7_GSEA_KEGG_GSEA/mmu03030.png             29
4 ./Project_D21_v_D7_GSEA_KEGG_GSEA/mmu03040.png             86
5 ./Project_D21_v_D7_GSEA_KEGG_GSEA/mmu04657.png             30
6 ./Project_D21_v_D7_GSEA_KEGG_GSEA/mmu04974.png             25
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       leadingEdgeId
1                                                                                                                                                                                                                                                                                                                                    12824;12845;12643;12841;21825;12840;21828;12839;12827;16779;20971;226519;12826;213119;12843;13138;16773;15530;12842;21827;15891;12834;16404;12833;11603;21826;319480;16398;12835;16776;81877;16419;14268;109700
2                                                                                                                                                                                                                                                                                                                     19192;56436;16912;26440;66997;70247;19177;22123;20422;23996;19166;19179;19167;19184;19173;17463;19186;69077;26443;66413;23997;21762;19172;66537;103554;19171;26442;19181;19170;19188;19185;26445;26441;26446;26444;19175;16913
3                                                                                                                                                                                                                                                                                                                                                                    19076;18968;18974;106344;68275;18972;19075;69724;68209;19687;18971;59001;72151;69263;17217;381760;67153;17220;17215;14156;19718;17218;68240;19891;17216;17219;18538;16881;66979
4 57905;20637;54633;66492;68879;20639;331401;72654;53317;320632;212880;66810;67039;68183;56031;72047;66354;56194;20624;60365;20222;22185;76522;81898;101943;66441;20643;230596;68092;13204;67465;66055;76938;68479;229279;20826;20638;68988;17149;56009;68011;27756;20630;75062;69878;107701;101214;56215;319322;66585;20382;19655;23983;73666;231889;107686;67959;67996;27366;51810;108014;20641;78651;66094;15381;108121;28000;68981;67332;15387;66101;53607;53817;76936;19656;20462;67678;68816;66125;66373;110809;50783;20383;15382;225027;21681
5                                                                                                                                                                                                                                                                                                                                                              14283;171095;17392;12370;71609;15519;15945;22033;14282;21929;16819;19225;22032;15568;18033;22027;16476;12367;110809;18035;12608;17395;20296;16176;20306;14825;20202;20201;20310;20311
6                                                                                                                                                                                                                                                                                                                                                                                             12824;12841;12815;12840;12839;12819;12827;12814;12826;12843;12842;13717;12834;11931;20514;12833;11933;73626;12822;12835;20540;11936;12816;69700;245026
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      userId
1                                                                                                                                                                                                                                                                                                                                            Agrn;Chad;Col1a1;Col1a2;Col2a1;Col4a1;Col4a2;Col6a1;Col6a2;Col6a3;Col9a1;Col9a2;Col9a3;Comp;Dag1;Fn1;Hspg2;Ibsp;Itga1;Itga10;Itga11;Itga2;Itga7;Itgb5;Lama2;Lama5;Lamb2;Lamc1;Sdc4;Thbs1;Thbs2;Thbs3;Thbs4;Tnxb
2                                                                                                                                                                                                                                                                                                                            Adrm1;Pomp;Psma1;Psma2;Psma3;Psma4;Psma5;Psma6;Psma7;Psmb1;Psmb10;Psmb2;Psmb3;Psmb4;Psmb5;Psmb6;Psmb7;Psmb8;Psmb9;Psmc1;Psmc2;Psmc4;Psmc5;Psmd1;Psmd11;Psmd12;Psmd13;Psmd2;Psmd3;Psmd4;Psmd6;Psmd7;Psme1;Psme2;Psme3;Psme4;Sem1
3                                                                                                                                                                                                                                                                                                                                                                                      Fen1;Lig1;Mcm2;Mcm3;Mcm4;Mcm5;Mcm6;Mcm7;Pcna;Pola1;Pold1;Pold2;Pole2;Pole3;Pole4;Prim1;Prim2;Rfc1;Rfc2;Rfc3;Rfc4;Rfc5;Rnaseh2a;Rnaseh2b;Rnaseh2c;Rpa1;Rpa2;Rpa3;Ssbp1
4 Acin1;Alyref;Alyref2;Bcas2;Bud31;Ccdc12;Ddx39b;Ddx42;Ddx46;Dhx15;Eftud2;Hnrnpa1;Hnrnpa3;Hnrnpc;Hnrnpk;Hnrnpm;Hnrnpu;Isy1;Lsm2;Lsm3;Lsm4;Lsm5;Lsm6;Lsm7;Lsm8;Magoh;Magohb;Ncbp2;Pcbp1;Phf5a;Plrg1;Ppie;Ppih;Ppil1;Pqbp1;Prpf19;Prpf31;Prpf38a;Prpf40a;Prpf6;Puf60;Rbm17;Rbm22;Rbm25;Rbm8a;Rbmx;Rbmxl1;Sf3a1;Sf3a2;Sf3a3;Sf3b1;Sf3b2;Sf3b3;Sf3b4;Sf3b5;Sf3b6;Snrnp200;Snrnp40;Snrnp70;Snrpa;Snrpa1;Snrpb;Snrpb2;Snrpc;Snrpd1;Snrpd2;Snrpd3;Snrpe;Snrpf;Snrpg;Snu13;Snw1;Srsf1;Srsf2;Srsf3;Srsf6;Srsf7;Srsf9;Thoc2;Thoc3;Tra2a;Tra2b;Txnl4a;U2af1;U2af2;Zmat2
5                                                                                                                                                                                                                                                                                                                                                                    Casp3;Casp8;Ccl2;Ccl7;Cebpb;Cxcl1;Cxcl10;Cxcl2;Cxcl5;Elavl1;Fosb;Fosl1;Hsp90aa1;Hsp90b1;Il17rc;Il1b;Jun;Lcn2;Mmp3;Mmp9;Nfkb1;Nfkbia;Ptgs2;S100a8;S100a9;Srsf1;Tnfaip3;Tradd;Traf4;Traf5
6                                                                                                                                                                                                                                                                                                                                                                    1810009J06Rik;Atp1b1;Atp1b3;Col11a1;Col11a2;Col12a1;Col15a1;Col18a1;Col1a1;Col1a2;Col22a1;Col2a1;Col4a1;Col4a2;Col6a1;Col6a2;Col6a3;Col6a6;Col9a1;Col9a2;Col9a3;Eln;Fxyd2;Slc1a5;Slc7a7</code></pre>
<p>The columns of the GSEA results are quite similar to the ORA results,
with the following exceptions:</p>
<ul>
<li><code>enrichmentScore</code>: reflects the degree to which a
gene-set is over-represented at the extremes of a ranked list of
genes.</li>
<li><code>normalizedEnrichmentScore</code>: corrects for differences in
ES between gene-sets due to differences in gene-set sizes to make the
scores of the tested gene-sets comparable with each other by dividing
the enrichment score (ES) by the mean of all ESs from all random
permutations</li>
<li><code>leadingEdgeNum</code>: is the subset of members that
contribute most to the ES; e.g. number of genes preceding the peak score
for positive enrichments.</li>
</ul>
<!-- TODO: Here is where we'll actually talk more about the biological relevance of the results. -->
<p>Additionally, for each pathway with <code>FDR &lt; fdrThr</code>, a
GSEA enrichment plot is automatically generated. The following is an
example for the ECM enrichment:</p>
<p><img
src="images/content/06-WebGestalt_GSEA/Day21_mmu04512_ECM.png" /></p>
<p>At the bottom, we can see the fold-change rankings of the query genes
from our comparison. In the middle are the marks indicating which genes
in the ranking are annotated for that pathway. At the top is the
enrichment score progression, with a dotted line at the max score
position.</p>
<div id="gsea-for-day-7-vs-day-0" class="section level2">
<h2>GSEA for Day 7 vs Day 0</h2>
<p>We’ll repeat the process for our Day 7 vs Day 0 comparison in
pericytes.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D7 vs D7 in pericyte cluster
isc_d7_v_d0 = read_csv(&#39;inputs/single_cell_de_results/de_pseudo_pericyte_D7_vs_D0.csv&#39;)</code></pre>
<pre><code>Rows: 22974 Columns: 6
── Column specification ───────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): gene
dbl (5): p_val, avg_log2FC, pct.1, pct.2, p_val_adj

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Preview the result
head(isc_d7_v_d0)</code></pre>
<pre><code># A tibble: 6 × 6
      p_val avg_log2FC pct.1 pct.2 p_val_adj gene   
      &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;  
1 3.98e-227      -5.49     1     1 1.05e-222 Cilp2  
2 6.29e-121      -3.78     1     1 1.67e-116 Tsc22d3
3 7.24e-121      -2.71     1     1 1.92e-116 Thbs4  
4 2.26e-119      -2.30     1     1 5.97e-115 Insc   
5 2.66e-118      -3.38     1     1 7.06e-114 Cst3   
6 3.92e-110      -3.72     1     1 1.04e-105 Ckb    </code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Select gene and avg_log2FC columns
isc_d7_v_d0_gsea = isc_d7_v_d0 %&gt;%
    dplyr::select(gene, avg_log2FC)

# -------------------------------------------------------------------------
# Preview the table
head(isc_d7_v_d0_gsea)</code></pre>
<pre><code># A tibble: 6 × 2
  gene    avg_log2FC
  &lt;chr&gt;        &lt;dbl&gt;
1 Cilp2        -5.49
2 Tsc22d3      -3.78
3 Thbs4        -2.71
4 Insc         -2.30
5 Cst3         -3.38
6 Ckb          -3.72</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Run GSEA
isc_d7_v_d0_gsea_result = WebGestaltR(
    enrichMethod = &#39;GSEA&#39;,
    nThreads = 8,
    organism = &#39;mmusculus&#39;,
    enrichDatabase = c(&#39;pathway_KEGG&#39;),
    interestGene = isc_d7_v_d0_gsea,
    interestGeneType = &#39;genesymbol&#39;,
    fdrThr = 0.1,
    outputDirectory = &#39;./results&#39;,
    projectName = &#39;D7_v_D0_GSEA_KEGG&#39;, 
    cache = NULL)</code></pre>
<pre><code>Loading the functional categories...
Loading the ID list...</code></pre>
<pre><code>Warning in dir.create(projectDir): &#39;./results/Project_D7_v_D0_GSEA_KEGG&#39; already
exists</code></pre>
<pre><code>Summarizing the uploaded ID list by GO Slim data...</code></pre>
<pre><code>Performing the enrichment analysis...</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00072</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00130</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00232</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00290</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00400</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00440</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00471</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00472</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00524</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00740</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00750</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00780</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00785</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00920</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu04122</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score has more than 500 items in set mmu01100</code></pre>
<pre><code>1000 permutations of score complete...</code></pre>
<pre><code>Begin affinity propagation...
End affinity propagation...
Begin weighted set cover...
End weighted set cover...
Generate the final report...
Results can be found in the ./results/Project_D7_v_D0_GSEA_KEGG!</code></pre>
<pre class="r"><code># clean up session
gc()</code></pre>
<pre><code>          used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells 2805963 149.9    5204824 278.0   5204824 278.0
Vcells 5954049  45.5   99769400 761.2 129823836 990.5</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# View the results for the other timepoint
head(isc_d7_v_d0_gsea_result)</code></pre>
<pre><code>   geneSet             description
1 mmu03010                Ribosome
2 mmu03030         DNA replication
3 mmu05134           Legionellosis
4 mmu05143 African trypanosomiasis
5 mmu04657 IL-17 signaling pathway
6 mmu03050              Proteasome
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       link
1 http://www.kegg.jp/kegg-bin/show_pathway?mmu03010+68028+19951+11837+20084+19896+27367+16785+20085+27050+66242+67945+16898+66480+19942+27207+67427+19944+20103+76808+68193+67115+270106+107734+94065+267019+76846+20104+20115+20068+67891+27397+22121+20116+19899+20091+19943+269261+26451+20044+66489+27370+20042+67941+56040+20055+19921+19988+78294+67025+68172+114641+20088+66483+68463+66475+68436+56282+68052+19981+54217+22186+68735+68611+64657+67248+54127+19934+19989+20102+100503670+67097+100043813+94063+26961+19941+14109+27398+67671+319195+107732+216767+66845+65019+19933+66407+67281+67186+19982+20005+118451+268449+20054+27176+67707+68537+66419+69527
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    http://www.kegg.jp/kegg-bin/show_pathway?mmu03030+66979+17215+68240+59001+19718+17219+67153+17216+69263+19891+18538+19076+14156+72151+16881+17218+17217+19687+18973+106344+18972+19075
3                                                                                                                                                                                                                                                                                                                                                                                                                                  http://www.kegg.jp/kegg-bin/show_pathway?mmu05134+14825+20310+18035+193740+67160+15511+16176+12367+12176+20224+16173+11783+94044+17874+66397+18033+16193+56422+76308+12475+13627+12369+11840+330122+269523+66824+12371+12362+13063+15510
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              http://www.kegg.jp/kegg-bin/show_pathway?mmu05143+15122+100503605+110257+18795+101488143+22329+18752+16176+16775+16173+18797
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                              http://www.kegg.jp/kegg-bin/show_pathway?mmu04657+20311+14825+20310+17386+17392+20201+20202+18035+19225+20306+16176+12367+17395+16476+22032+14282+50772+15945+21929+12608+110809+14283+15519
6                                                                                                                                                                                                                                                                                                                                                                                                                                                                        http://www.kegg.jp/kegg-bin/show_pathway?mmu03050+16913+26444+26445+19170+20422+57296+26446+66413+19172+56436+19171+19175+26442+103554+19192+59029+21762+19182+22123+19173+26441+19185+19186+19167
  enrichmentScore normalizedEnrichmentScore      pValue         FDR size
1       0.7912312                  2.414017 0.000000000 0.000000000  129
2       0.7655349                  1.876325 0.000000000 0.004209260   35
3       0.6762411                  1.808611 0.000000000 0.007215875   56
4       0.7320667                  1.811441 0.000000000 0.009220285   33
5       0.6016039                  1.771046 0.000000000 0.012507517   86
6       0.6611633                  1.715198 0.004618938 0.021196633   46
                                       plotPath leadingEdgeNum
1 ./Project_D7_v_D0_GSEA_KEGG_GSEA/mmu03010.png             97
2 ./Project_D7_v_D0_GSEA_KEGG_GSEA/mmu03030.png             22
3 ./Project_D7_v_D0_GSEA_KEGG_GSEA/mmu05134.png             30
4 ./Project_D7_v_D0_GSEA_KEGG_GSEA/mmu05143.png             11
5 ./Project_D7_v_D0_GSEA_KEGG_GSEA/mmu04657.png             23
6 ./Project_D7_v_D0_GSEA_KEGG_GSEA/mmu03050.png             24
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            leadingEdgeId
1 68028;19951;11837;20084;19896;27367;16785;20085;27050;66242;67945;16898;66480;19942;27207;67427;19944;20103;76808;68193;67115;270106;107734;94065;267019;76846;20104;20115;20068;67891;27397;22121;20116;19899;20091;19943;269261;26451;20044;66489;27370;20042;67941;56040;20055;19921;19988;78294;67025;68172;114641;20088;66483;68463;66475;68436;56282;68052;19981;54217;22186;68735;68611;64657;67248;54127;19934;19989;20102;100503670;67097;100043813;94063;26961;19941;14109;27398;67671;319195;107732;216767;66845;65019;19933;66407;67281;67186;19982;20005;118451;268449;20054;27176;67707;68537;66419;69527
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    66979;17215;68240;59001;19718;17219;67153;17216;69263;19891;18538;19076;14156;72151;16881;17218;17217;19687;18973;106344;18972;19075
3                                                                                                                                                                                                                                                                                                                                                                                                                                  14825;20310;18035;193740;67160;15511;16176;12367;12176;20224;16173;11783;94044;17874;66397;18033;16193;56422;76308;12475;13627;12369;11840;330122;269523;66824;12371;12362;13063;15510
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              15122;100503605;110257;18795;101488143;22329;18752;16176;16775;16173;18797
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                              20311;14825;20310;17386;17392;20201;20202;18035;19225;20306;16176;12367;17395;16476;22032;14282;50772;15945;21929;12608;110809;14283;15519
6                                                                                                                                                                                                                                                                                                                                                                                                                                                                        16913;26444;26445;19170;20422;57296;26446;66413;19172;56436;19171;19175;26442;103554;19192;59029;21762;19182;22123;19173;26441;19185;19186;19167
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    userId
1 Fau;Mrpl10;Mrpl11;Mrpl12;Mrpl13;Mrpl14;Mrpl16;Mrpl17;Mrpl2;Mrpl22;Mrpl24;Mrpl28;Mrpl30;Mrpl33;Mrpl34;Mrps10;Mrps15;Mrps16;Mrps18c;Mrps2;Mrps9;Rpl10a;Rpl11;Rpl12;Rpl13;Rpl13a;Rpl14;Rpl15;Rpl17;Rpl18;Rpl18a;Rpl19;Rpl21;Rpl22;Rpl22l1;Rpl23;Rpl23a;Rpl24;Rpl26;Rpl27;Rpl27a;Rpl28;Rpl29;Rpl3;Rpl31;Rpl32;Rpl34;Rpl35;Rpl36;Rpl36a;Rpl36al;Rpl37;Rpl37a;Rpl38;Rpl39;Rpl39l;Rpl4;Rpl41;Rpl5;Rpl6;Rpl7;Rpl7a;Rpl8;Rpl9;Rplp0;Rplp1;Rplp2;Rps10;Rps11;Rps12;Rps13;Rps14;Rps15;Rps15a;Rps16;Rps17;Rps18;Rps19;Rps2;Rps20;Rps23;Rps24;Rps26;Rps27a;Rps27l;Rps27rt;Rps28;Rps3;Rps3a1;Rps4x;Rps5;Rps6;Rps7;Rps8;Rps9;Rpsa;Uba52
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   Fen1;Lig1;Mcm2;Mcm3;Mcm4;Mcm5;Mcm6;Pcna;Pold2;Pole;Pole3;Pole4;Prim1;Prim2;Rfc1;Rfc2;Rfc3;Rfc4;Rfc5;Rnaseh2b;Rpa2;Rpa3
3                                                                                                                                                                                                                                                                                                                                                                                                                                        Apaf1;Arf1;Bcl2l13;Bnip3;Casp1;Casp3;Casp7;Casp9;Cd14;Cxcl1;Cxcl2;Cxcl3;Cycs;Eef1a1;Eef1g;Hbs1l;Hspa1a;Hspa1b;Hspd1;Il18;Il1b;Il6;Myd88;Nfkb1;Nfkbia;Pycard;Rab1b;Sar1a;Sar1b;Vcp
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      Hba-a1;Hba-a2;Hbb-bs;Hbb-bt;Il18;Il1b;Lama4;Plcb1;Plcb3;Prkcg;Vcam1
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Casp3;Ccl7;Cebpb;Cxcl1;Cxcl10;Cxcl2;Cxcl5;Fosb;Fosl1;Hsp90aa1;Il1b;Jun;Mapk6;Mmp13;Mmp3;Mmp9;Nfkbia;Ptgs2;S100a8;S100a9;Srsf1;Tnfaip3;Traf4
6                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Adrm1;Psma3;Psma4;Psma5;Psma7;Psmb1;Psmb10;Psmb2;Psmb3;Psmb4;Psmb5;Psmb6;Psmb8;Psmc3;Psmd14;Psmd2;Psmd3;Psmd4;Psmd6;Psmd8;Psme1;Psme3;Psme4;Sem1</code></pre>
<p>Here we see positive enrichment scores</p>
</div>
</div>
<div id="comparing-enrichment-results" class="section level1">
<h1>Comparing enrichment results</h1>
<p>Now that we have two GSEA results from related differential
expression results, we may want to know:</p>
<ul>
<li>What pathways are shared versus unique between the timepoints?</li>
<li>Are there any insights we can glean by comparing pathways at one
time point versus another?</li>
</ul>
<p>We’ll use some of the advanced visualizations from yesterday to
answer some of these questions.</p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<ul>
<li>GSEA is a functional enrichment choice that does not require using
thresholds to define genes of interest</li>
<li>KEGG pathways offers an alternative knowledge base that can be
useful for interpretation</li>
</ul>
</div>
<div id="questions" class="section level1">
<h1>Questions</h1>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"><a href="05-fcs-gsea-overview.html">Previous
lesson</a></th>
<th align="center"><a href="#top">Top of this lesson</a></th>
<th align="right"><a href="07B-advanced-visualizations.html">Next
lesson</a></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
