<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Analysis Summary</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link rel="shortcut icon" href="favicon-16x16.png" />
<!--
Favicon from built via
https://favicon.io/emoji-favicons/computer/
-->
<meta name="google-site-verification" content="t-Wgs-RXt8xi94RSoFamxoD2P2h-Hh7M_t_iOuxFZUo" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intro Functional Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="workshop-intro.html">Intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-overview.pdf">Functional analysis and ORA</a>
    </li>
    <li>
      <a href="02-IntroToWebGestaltandORA.html">ORA with WebGestalt</a>
    </li>
    <li>
      <a href="03-WebGestaltRORA.html">ORA with R</a>
    </li>
    <li>
      <a href="07A-advanced-visualizations.html">Advanced visualizations I</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-gsea-overview.pdf">Functional Analysis and GSEA</a>
    </li>
    <li>
      <a href="06-WebGestaltRGSEA.html">GSEA with R</a>
    </li>
    <li>
      <a href="gene-set-references.pdf">GO, KEGG, and friends</a>
    </li>
    <li>
      <a href="07B-advanced-visualizations.html">Advanced visualizations II</a>
    </li>
    <li>
      <a href="08-analysis-summary.html">Analysis summary</a>
    </li>
  </ul>
</li>
<li>
  <a href="workshop-wrap-up.html">Wrap up</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis Summary</h1>

</div>


<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>
<script>
  addClassKlippyTo("pre.r, pre.markdown, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<p><img src="images/wayfinder/wayfinder-wrap-up.png"
alt="workshop wayfinder" /></p>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>Functional analysis can be a powerful tool for summarizing
differential expression results from a variety of experiments, including
both bulk and single-cell RNA-seq.</p>
<p>What we’ve accomplished:</p>
<ul>
<li>Discussed the concepts behind ORA, GSEA, and Pathway Topology
approaches to functional enrichment approaches.</li>
<li>Used a web-browser tool to perform ORA analysis of bulk RNA-seq
data.<br />
</li>
<li>Used R/RStudio and <code>WebGestaltR</code> to run ORA for bulk
RNA-seq data in a more reproducible manner.<br />
</li>
<li>Used <code>WebGestaltR</code> to run GSEA for single-cell RNA-seq
data<br />
</li>
<li>Created customized visualizations at the gene and term level<br />
</li>
<li>Reviewed biological knowledge bases commonly used in functional
analyses.</li>
</ul>
<p><br></p>
<hr />
</div>
<div id="resources" class="section level1">
<h1>Resources</h1>
<div id="methods-and-approaches" class="section level2">
<h2>Methods and Approaches</h2>
<ul>
<li><p>Khatri P, Sirota M, Butte AJ. Ten years of pathway analysis:
current approaches and outstanding challenges. PLoS Comput Biol.
2012;8(2):e1002375. <a href="doi:10.1371/journal.pcbi.1002375"
class="uri">doi:10.1371/journal.pcbi.1002375</a> <a
href="https://pubmed.ncbi.nlm.nih.gov/22383865/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/22383865/</a></p></li>
<li><p>Subramanian A, Tamayo P, Mootha VK, et al. Gene set enrichment
analysis: a knowledge-based approach for interpreting genome-wide
expression profiles. Proc Natl Acad Sci U S A. 2005;102(43):15545-15550.
<a href="doi:10.1073/pnas.0506580102"
class="uri">doi:10.1073/pnas.0506580102</a><br />
<a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC1239896/"
class="uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC1239896/</a></p></li>
<li><p>Wijesooriya K, Jadaan SA, Perera KL, Kaur T, Ziemann M. Urgent
need for consistent standards in functional enrichment analysis. PLoS
Comput Biol. 2022;18(3):e1009935. Published 2022 Mar 9. <a
href="doi:10.1371/journal.pcbi.1009935"
class="uri">doi:10.1371/journal.pcbi.1009935</a><br />
<a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8936487/"
class="uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC8936487/</a></p></li>
<li><p>Zyla J, Marczyk M, Weiner J, Polanska J. Ranking metrics in gene
set enrichment analysis: do they matter?. BMC Bioinformatics.
2017;18(1):256. Published 2017 May 12. <a
href="doi:10.1186/s12859-017-1674-0"
class="uri">doi:10.1186/s12859-017-1674-0</a> <a
href="https://pubmed.ncbi.nlm.nih.gov/28499413/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/28499413/</a></p></li>
<li><p>Topology Pathway Analyses</p>
<ul>
<li>Tarca AL, Draghici S, Khatri P, et al. A novel signaling pathway
impact analysis. Bioinformatics. 2009;25(1):75-82. <a
href="doi:10.1093/bioinformatics/btn577"
class="uri">doi:10.1093/bioinformatics/btn577</a><br />
<a href="https://pubmed.ncbi.nlm.nih.gov/18990722/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/18990722/</a></li>
<li>Nguyen T, Mitrea C, Draghici S. Network-Based Approaches for Pathway
Level Analysis. Curr Protoc Bioinformatics.
2018;61(1):8.25.1-8.25.24.<br />
<a href="doi:10.1002/cpbi.42" class="uri">doi:10.1002/cpbi.42</a><br />
<a href="https://pubmed.ncbi.nlm.nih.gov/30040185/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/30040185/</a></li>
<li>Ma J, Shojaie A, Michailidis G. A comparative study of
topology-based pathway enrichment analysis methods. BMC Bioinformatics.
2019;20(1):546. Published 2019 Nov 4. <a
href="doi:10.1186/s12859-019-3146-1"
class="uri">doi:10.1186/s12859-019-3146-1</a> <a
href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3146-1"
class="uri">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-3146-1</a></li>
<li>Ihnatova I, Popovici V, Budinska E. A critical comparison of
topology-based pathway analysis methods. PLoS One. 2018;13(1):e0191154.
Published 2018 Jan
<ol start="25" style="list-style-type: decimal">
<li><a href="doi:10.1371/journal.pone.0191154"
class="uri">doi:10.1371/journal.pone.0191154</a>.<br />
<a href="https://pubmed.ncbi.nlm.nih.gov/29370226/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/29370226/</a></li>
</ol></li>
</ul></li>
</ul>
</div>
<div id="biological-knowledge-bases" class="section level2">
<h2>Biological Knowledge Bases</h2>
<ul>
<li>Gene Ontology (GO)
<ul>
<li><a href="https://geneontology.org/docs/ontology-documentation/"
class="uri">https://geneontology.org/docs/ontology-documentation/</a></li>
<li><a href="https://geneontology.org/stats.html"
class="uri">https://geneontology.org/stats.html</a></li>
<li>du Plessis L, Skunca N, Dessimoz C. The what, where, how and why of
gene ontology–a primer for bioinformaticians. Brief Bioinform.
2011;12(6):723-735. <a href="doi:10.1093/bib/bbr002"
class="uri">doi:10.1093/bib/bbr002</a> <a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC3220872/"
class="uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC3220872/</a></li>
</ul></li>
<li>KEGG
<ul>
<li><a href="https://www.genome.jp/kegg/kegg1a.html"
class="uri">https://www.genome.jp/kegg/kegg1a.html</a></li>
<li><a href="https://www.genome.jp/kegg/docs/statistics.html"
class="uri">https://www.genome.jp/kegg/docs/statistics.html</a></li>
</ul></li>
<li>Reactome
<ul>
<li><a href="https://reactome.org/"
class="uri">https://reactome.org/</a></li>
</ul></li>
<li>MSigDB
<ul>
<li><a href="https://www.gsea-msigdb.org/gsea/msigdb"
class="uri">https://www.gsea-msigdb.org/gsea/msigdb</a></li>
<li>Liberzon A, Birger C, Thorvaldsdóttir H, Ghandi M, Mesirov JP,
Tamayo P. The Molecular Signatures Database (MSigDB) hallmark gene set
collection. Cell Syst. 2015;1(6):417-425. <a
href="doi:10.1016/j.cels.2015.12.004"
class="uri">doi:10.1016/j.cels.2015.12.004</a><br />
<a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC4707969/"
class="uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC4707969/</a></li>
</ul></li>
</ul>
</div>
<div id="analysis-tools" class="section level2">
<h2>Analysis tools</h2>
<ul>
<li><a href="https://www.webgestalt.org/">WebGestalt</a>
<ul>
<li>Wang J, Vasaikar S, Shi Z, Greer M, Zhang B. WebGestalt 2017: a more
comprehensive, powerful, flexible and interactive gene set enrichment
analysis toolkit. Nucleic Acids Res. 2017;45(W1):W130-W137. <a
href="doi:10.1093/nar/gkx356" class="uri">doi:10.1093/nar/gkx356</a> <a
href="https://pubmed.ncbi.nlm.nih.gov/28472511/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/28472511/</a></li>
<li>Elizarraras JM, Liao Y, Shi Z, Zhu Q, Pico AR, Zhang B. WebGestalt
2024: faster gene set analysis and new support for metabolomics and
multi-omics. Nucleic Acids Res. 2024;52(W1):W415-W421. <a
href="doi:10.1093/nar/gkae456" class="uri">doi:10.1093/nar/gkae456</a>
<a href="https://pubmed.ncbi.nlm.nih.gov/38808672/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/38808672/</a></li>
</ul></li>
<li><a href="https://www.gsea-msigdb.org/gsea/index.jsp">GSEA</a>
<ul>
<li><a
href="https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideFrame.html">Broad
GSEA User Guide</a></li>
</ul></li>
<li><a
href="https://www.bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a></li>
<li><a
href="https://bioconductor.org/packages/devel/bioc/vignettes/fgsea/inst/doc/fgsea-tutorial.html">fGSEA</a></li>
<li><a href="https://davidbioinformatics.nih.gov/">DAVID</a></li>
<li><a
href="https://advaitabio.com/bioinformatics/ipathwayguide/">iPathwayGuide</a></li>
<li>Nice for proteomic data:
<ul>
<li><a href="https://varemo.github.io/piano/">Piano</a></li>
<li><a href="https://string-db.org/">String</a></li>
</ul></li>
</ul>
</div>
<div id="tutorials" class="section level2">
<h2>Tutorials</h2>
<ul>
<li><a
href="https://hbctraining.github.io/Training-modules/DGE-functional-analysis/lessons/functional_analysis_2019.html">Harvard
Chan Bioinformatics Core on Functional Analysis</a></li>
<li><a
href="https://diytranscriptomics.com/project/lecture-10">DIY.transcriptomics
Functional enrichment analysis</a></li>
<li><a
href="https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/">GSEA
with ClusterProfiler</a></li>
<li><a
href="https://www.borch.dev/uploads/screpertoire/articles/running_escape">ESCAPE
for single-cell enrichments</a></li>
<li><a
href="https://www.sc-best-practices.org/conditions/gsea_pathway.html">Single
Cell Best Practices book chapter on Funtional Enrichments</a></li>
<li>Enrichment visualization with <a
href="https://yulab-smu.top/biomedical-knowledge-mining-book/enrichplot.html">enrichplot</a></li>
<li><a href="https://biostatsquid.com/">BiostatsSquid</a> on <a
href="https://www.youtube.com/watch?v=H1cUs6pql9s">Pathway Enrichment
Analysis</a> and <a
href="https://www.youtube.com/watch?v=egO7Lt92gDY">GSEA</a></li>
</ul>
<br/>
<hr/>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"><a href="07B-advanced-visualizations.html">Previous
lesson</a></th>
<th align="center"><a href="#top">Top of this lesson</a></th>
<th align="right"><a href="workshop-wrap-up.html">Next lesson</a></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
