---
title: "GSEA with R"
output:
        html_document:
            includes:
                in_header: header.html
            theme: paper
            toc: true
            toc_depth: 4
            toc_float: true
            number_sections: false
            fig_caption: false
            markdown: GFM
            code_download: false
---

<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>

```{r klippy, echo=FALSE, include=TRUE}
klippy::klippy(lang = c("r", "markdown", "bash"), position = c("top", "right"))
```

```{r, 'chunk_options', include=FALSE}
source("../bin/chunk-options.R")
knitr_fig_path("02-")
```

```{r, include = FALSE}
source("../bin/set_values.R")
```

![workshop wayfinder](images/wayfinder/wayfinder-06A.png)

# Objectives

- Form the inputs for Gene Set Enrichment Analysis (GSEA) in WebGestaltR from single-cell results.
- Use WebGestaltR to do a GSEA analysis.
- Understand how to interpret the results of GSEA analysis.
- Compare and contrast the results of two GSEA results.

# Getting started

The setup for GSEA is similar to ORA, but there are two key differences:

1. We do not require a threshold for significance, and
2. We need to pass gene identifiers **and** the log fold-change for the gene from the differential expression test.

This gives the advantage of being able to use the differential expression results more fully, instead of limiting to DE genes.

# Running GSEA using WebGestalt

First, we'll re-load the libraries to ensure they are available.

```{r load_webgestalt}
# =========================================================================
# ORA with WebGestlatR
# =========================================================================

# -------------------------------------------------------------------------
# Load the libraries
library(WebGestaltR)
library(tidyverse)
```


We'll then will take a second look at the documentation for the `WebGestaltR()` function, and note the difference in input between ORA and GSEA.

```{r webgestalt_doc, eval = FALSE}
# -------------------------------------------------------------------------
# Look at the manual for the main function
?WebGestaltR
```

We note from the documentation that `interestGene` "should be an R data.frame object containing two columns: the gene list and the corresponding scores." We also note that we will **not** make use of the `referenceGene` argument.

## Read in diffex results

We will demonstrate GSEA on a differential analysis from the single-cell RNA-seq experiment we explored in the Introduction to Single-cell Analysis [(link)](https://umich-brcf-bioinf.github.io/workshop-intro-single-cell/main/html/00A-OrientingOnScRNASeq.html#Consider_a_specific_scRNA-Seq_experiment). In particular, we'll perform GSEA on the differential expression results in the pericyte cluster, starting with the results from the day 21 to day 7 comparison.

```{r isc_read}
# -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D21 vs D7 in pericyte cluster
isc_d21_v_d7 = read_csv('inputs/single_cell_de_results/de_pseudo_pericyte_D21_vs_D7.csv')

# -------------------------------------------------------------------------
# Preview the result
head(isc_d21_v_d7)
```

Before preparing our data for GSEA, let's also check how many genes were DE in the pseudobulk D21 vs D7 comparison using the same thresholds from the workshop. 
```{r isc_DE}
# -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D21 vs D7 in pericyte cluster
table(isc_d21_v_d7$p_val_adj < 0.05 & abs(isc_d21_v_d7$avg_log2FC) > 1.5)

```

<!-- NOTE - mistake in ISC results - miss log2 transformation of threshold in table step (https://umich-brcf-bioinf.github.io/workshop-intro-single-cell/main/html/08-DifferentialExpression.html#Pseudobulk_comparisons) so the number of DE genes artificially  deflated and actually # of DE genes for scRNA is ~400 so is likely a better candidate for the ORA over the bulk results -->

## Extracting data for GSEA

The key columns to extract from our Day 21 vs Day 7 pericyte results table, as per the documentation, are the `gene` and a column of values to use to rank the genes. Before proceeding, it's prudent to check if there are missing values in the column we'll use as our gene identifier:

```{r isc_na}
# -------------------------------------------------------------------------
# How many symbols are NA?
table(is.na(isc_d21_v_d7$gene))
```

For ranking, we can use the values from the `avg_log2FC` column and create a table with just the `gene` and `avg_log2FC` columns with the `select()` function from `dplyr`:

```{r isc_interest_set}
# -------------------------------------------------------------------------
# Select gene and avg_log2FC columns
isc_d21_v_d7_gsea = isc_d21_v_d7 %>%
    dplyr::select(gene, avg_log2FC)

# -------------------------------------------------------------------------
# Preview the table
head(isc_d21_v_d7_gsea)
```


# Running GSEA

We will start by running GSEA on the non-redundant version of the Gene Ontology Biological Process genesets.

```{r isc_gsea}
# -------------------------------------------------------------------------
# Run GSEA
isc_d21_v_d7_gsea_result = WebGestaltR(
    enrichMethod = 'GSEA',
    nThreads = 8,
    organism = 'mmusculus',
    enrichDatabase = c('pathway_KEGG'),
    interestGene = isc_d21_v_d7_gsea,
    interestGeneType = 'genesymbol',
    fdrThr = 0.1,
    outputDirectory = './results',
    projectName = 'D21_v_D7_GSEA_KEGG', 
    cache = NULL)

# clean up session
gc()
```
One downside of GSEA is that it is more computationally demanding than ORA. While the GSEA analysis is running, let's touch on why this enrichment process takes so much longer. As the enrichment runs, we'll see the reporting messages `1000 permutations of score complete` this permutation was introduced in the overview and is what generates the null distribution that's needed to compare against to answer the question of "how surprising".

But what about our input? Why did we provide fold-change as a column for ranking our genes?

<!-- what to add here?-->

For more details, this [Pathways Common guide](https://www.pathwaycommons.org/guide/primers/data_analysis/gsea/) delves into the history and algorithm details 


<!-- These results are from the genes ranked by effect size only (e.g. fold-change). What would change if we incorporated both the sign of the fold change and the level of significance? 

Spoiler - both -log10pvalue * logFc or * sign of logFC yeild no sig enrichment above threhsold

```{r isc_interest_set2}
# -------------------------------------------------------------------------
# Select gene and combined stat column that will be used for ranking
isc_d21_v_d7_gsea_p2 = isc_d21_v_d7 %>%
  dplyr::mutate(pres = -log10(p_val_adj) * sign(avg_log2FC)) %>%
  dplyr::select(gene, pres) 

# -------------------------------------------------------------------------
# Preview the table
head(isc_d21_v_d7_gsea_p2)

```


```{r isc_gsea_2}
# -------------------------------------------------------------------------
# Run GSEA
isc_d21_v_d7_gsea_result_2 = WebGestaltR(
    enrichMethod = 'GSEA',
    nThreads = 8,
    organism = 'mmusculus',
    enrichDatabase = c('pathway_KEGG'),
    interestGene = isc_d21_v_d7_gsea_p2,
    interestGeneType = 'genesymbol',
    fdrThr = 0.1,
    outputDirectory = './results',
    projectName = 'D21_v_D7_GSEA_KEGG_alt', 
    cache = NULL)

# clean up session
gc()

# look at results
isc_d21_v_d7_gsea_result_2
```
--->


```{r isc_check_results}
# -------------------------------------------------------------------------
# View the first few results
head(isc_d21_v_d7_gsea_result)
```

The columns of the GSEA results are quite similar to the ORA results, with the following exceptions:

- `enrichmentScore`: 
- `normalizedEnrichmentScore`:
- `leadingEdgeNum`: 

<!-- TODO: Here is where we'll actually talk more about the biological relevance of the results. -->

# Results of GSEA

As with the ORA results, the GSEA method outputs similar results tables, gene identifier mappings, and HTML reports. Additionally, for each pathway with `FDR < fdrThr`, a GSEA enrichment plot is automatically generated. The following is an example:

TODO: Include example GSEA plot.


![workshop wayfinder](images/wayfinder/wayfinder-06B.png)


# Comparing enrichment results

Now that we have two GSEA results from related differential expression results, we may want to know:

- What genesets are in common?
- What genesets are unique?
- Are there any insights in the pathways at one time point versus another?

A first approach to these questions might be to take the GSEA results from both comparisons and join them on the geneset identifier.



# Summary

# Questions


| [Previous lesson](05-fcs-gsea-overview.html) | [Top of this lesson](#top) | [Next lesson](04-gene-set-references.html) |
| :--- | :----: | ---: |