---
title: "Advanced Visualizations II"
output:
        html_document:
            includes:
                in_header: header.html
            theme: paper
            toc: true
            toc_depth: 4
            toc_float: true
            number_sections: false
            fig_caption: false
            markdown: GFM
            code_download: false
---

<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>

```{r klippy, echo=FALSE, include=TRUE}
klippy::klippy(lang = c("r", "markdown", "bash"), position = c("top", "right"))
```

```{r, 'chunk_options', include=FALSE}
source("../bin/chunk-options.R")
knitr_fig_path("02-")
```

![workshop wayfinder](images/wayfinder/wayfinder-07B.png)

# Objectives

- Compare significant pathways between multiple enrichment results
- Highlight some interesting pathways from the meta-analysis


# Getting started

<!-- # library(VennDiagram) # missing library -->
```{r libraries}
# =========================================================================
# Advanced Visualizations II
# =========================================================================

# -------------------------------------------------------------------------
# Load additional libraries
library(ComplexHeatmap)
library(ggVennDiagram)

# -------------------------------------------------------------------------
# Check current working directory
getwd()
```

We should have the GSEA results in our session but we can read in the results from the table output to file by `WebGestaltR` for the results from the Day21 vs Day7 and Day7 vs Day 0 pericyte single-cell RNA-seq comparisons. 

```{r viz_read_ora}
# -------------------------------------------------------------------------
# Read in results for the enrichment for each set of timpoint comparisons
isc_d21_v_d7_gsea_result = read_delim('results/Project_D21_v_D7_GSEA_KEGG/enrichment_results_D21_v_D7_GSEA_KEGG.txt')

isc_d7_v_d0_gsea_result = read_delim('results/Project_D7_v_D0_GSEA_KEGG//enrichment_results_D7_v_D0_GSEA_KEGG.txt')

head(isc_d21_v_d7_gsea_result); head(isc_d7_v_d0_gsea_result)
```
<br>

Instead of comparing gene sets between enriched terms from a single set of results, we can instead compare enrichments between two different comparisons. Since the KEGG ids and descriptions are organized into columns of the results tables we don't need to do as much processing to compare between the enriched pathways. 

```{r path_overlaps}
# -------------------------------------------------------------------------
# Create list of enriched KEGG pathways 
KEGG_results <- c()
KEGG_results[["d7_v_d0"]] = isc_d7_v_d0_gsea_result$description
KEGG_results[["d21_v_d7"]] = isc_d21_v_d7_gsea_result$description
```

Once we've created the lists of enriched pathways, we can use operations like `union` and `intersect` to compare between the sets. 

```{r path_compare}
# -------------------------------------------------------------------------
# Identify total number of unique pathways enriched across both timepoints
union(KEGG_results[["d7_v_d0"]], KEGG_results[["d21_v_d7"]]) %>% length()

# -------------------------------------------------------------------------
# Identify number of pathways shared between timepoints
intersect(KEGG_results[["d7_v_d0"]], KEGG_results[["d21_v_d7"]]) %>% length()
```


Since the number of sets in one of our comparisons exceeds the limit allowed for `ComplexHeatmap`'s version of an upSet plot (which is `31`), we'll create a venn diagram comparing the number of shared/unique pathways for each timepoint using the `ggVennDiagram` option, which is [one of many options available to create this kind of plot](https://www.datanovia.com/en/blog/beautiful-ggplot-venn-diagram-with-r), instead.


```{r path_Venn, eval=TRUE}
# -------------------------------------------------------------------------
# Make simple plot to compare KEGG pathways enriched for each timepoint
KEGG_venn <- ggVennDiagram(KEGG_results, label_alpha = 0, 
                           label_size = 6, set_size = 6,
                           ) +
  ggplot2::scale_fill_gradient(low="royalblue",high = "yellow") +
  coord_flip()

KEGG_venn
```
<br>

Then we can output the plot to file:

```{r output_venn}
# -------------------------------------------------------------------------
# Output UpSet plot to file
png(file = "./results/figures/Pericytes_KEGG_vennDiagram.png", width = 600, height = 400)
  KEGG_venn # draw the plot
dev.off() # Close the graphics device
```


# Detailed comparison of KEGG enrichments

From the Venn diagram, we have a summary of the number of KEGG pathways that are shared or uniquely enriched for the two sets of comparisons from the pericyte population but that doesn't tell us the magnitudes of those enrichments or if they were in the same direction or opposite direction between the earlier timepoint comparison and later timepoint comparison. 

<!-- ideally add EnrichR and/or ClusterProfiler packages since have improved visualizations compared to  -->

```{r combine_KEGG}
# -------------------------------------------------------------------------
# Create 'tidy' version  of Day 7 vs Day 0 GSEA results
isc_d7_v_d0_gsea_simplified <- isc_d7_v_d0_gsea_result %>% 
  select(geneSet, description, normalizedEnrichmentScore, size, FDR) %>% 
  pivot_longer(cols = !c(geneSet, description, FDR, size)) %>% 
  mutate(timepoint = "D7_v_D0", .before = 1)

# -------------------------------------------------------------------------
# Create 'tidy' version  of Day 21 vs Day 7 GSEA results
isc_d21_v_d7_gsea_simplified <- isc_d21_v_d7_gsea_result %>% 
  select(geneSet, description, normalizedEnrichmentScore, size, FDR) %>% 
  pivot_longer(cols = !c(geneSet, description, FDR, size)) %>% 
  mutate(timepoint = "D21_v_D7", .before = 1)

# -------------------------------------------------------------------------
# Combine 'tidy' results into single table
KEGG_results_combined <- rbind(isc_d7_v_d0_gsea_simplified, isc_d21_v_d7_gsea_simplified) 
KEGG_results_combined <- KEGG_results_combined %>% 
  mutate(timepoint = factor(timepoint, levels=c("D7_v_D0", "D21_v_D7")),
         direction = factor(ifelse(value < 0, "negative NES", "positive NES")),
         "positive NES", "negative NES") # fix levels & add column for direction

# -------------------------------------------------------------------------
# Check table
str(KEGG_results_combined)
```

Now that we have a nice "tidy" version of the enrichment results, we can use it to create a barplot that summarizes the enrichment scores for each timepoint.

```{r plot_KEGG_simple}
# -------------------------------------------------------------------------
# Graph simple KEGG pathway enrichments for both timepoints
KEGG_comparison <- ggplot(KEGG_results_combined, 
                          aes(y=value, x=reorder(description, value))) + 
  geom_bar(position="dodge", stat="identity") +
  facet_wrap(~timepoint) +
  xlab("") + coord_flip()
KEGG_comparison
```

<br>

While this barplot has some interesting patterns, it's a bit hard to read and isn't very pretty. 

Let's create a modified version that filters our enrichment results and adds attributes to the plot to show some key metrics, including the direction and magnitude of the normalized enrichment score, the significance, and the number of genes in the pathway (size). 

```{r plot_KEGG_complex}
# -------------------------------------------------------------------------
# Subset to results that pass more stringent results
KEGG_results_strict <- KEGG_results_combined %>% filter(FDR < 0.025 & size > 60)

# -------------------------------------------------------------------------
# Graph simple KEGG pathway enrichments for both timepoints
KEGG_comparison <- ggplot(KEGG_results_strict, aes(y=value, x=reorder(description, value))) + 
  geom_segment(aes(y = 0, x = description, yend = value, xend = description),
               color = ifelse(KEGG_results_strict$direction %in% c("positive NES"), "red", "blue"),
               linewidth = 1) +
  geom_point(aes(color=FDR, size = size), stat="identity") +
  scale_color_continuous(low = "black", high = "grey") +
  facet_wrap(~timepoint) +
  geom_hline(yintercept=0, linetype="dashed", color = "lightgrey") +
  labs(title="KEGG enrichments in pericytes") +
  xlab("") + theme_bw() +
  coord_flip()
KEGG_comparison
```
<br>

What we created here is called a "lollipop" plot, which we'll save to file.

```{r save_lollipop}
# -------------------------------------------------------------------------
# Output UpSet plot to file
png(file = "./results/figures/Pericytes_KEGG_lollipopComparison.png", width = 700, height = 600, res = 100)
  KEGG_comparison # draw the plot
dev.off() # Close the graphics device
```
<br>

From the comparison of enriched KEGG pathways between the timepoints, one of the results that jumps out is the the positive enrichment score for "IL-17 signaling pathway" at the early time points comparison and the negative enrichment score at the later time points comparison. IL-17 signalling has a [role in wound repair](https://pmc.ncbi.nlm.nih.gov/articles/PMC11129447/) so it's activation at the early time point and suppression at the later time point fits with our general expectations but might be interesting depending on what's known about IL-17 signaling in pericytes.  

Another result that jumps out is the enrichment only at the later timepoint of "ECM-receptor interaction", which has a [role in wound healing](https://pmc.ncbi.nlm.nih.gov/articles/PMC9326521/) but also [bone formation](https://pmc.ncbi.nlm.nih.gov/articles/PMC7264100/). 

<br>

<!-- How did we decide to make this kind of plot and what did we do to create it? Add steps of search/iteration?
Links:
- https://r-graph-gallery.com/48-grouped-barplot-with-ggplot2.html
- https://r-charts.com/part-whole/diverging-bar-chart-ggplot2/
- https://www.tutorialspoint.com/ggplot2/ggplot2_diverging_charts.html
- https://yulab-smu.top/biomedical-knowledge-mining-book/enrichplot.html
- https://stackoverflow.com/questions/15116081/controlling-order-of-facet-grid-facet-wrap-in-ggplot2
- https://r-graph-gallery.com/304-highlight-a-group-in-lollipop.html
- https://ggplot2-book.org/scales-colour.html#sec-binned-colour
- https://www.sthda.com/english/wiki/ggplot2-colors-how-to-change-colors-automatically-and-manually
- https://stackoverflow.com/questions/15116081/controlling-order-of-facet-grid-facet-wrap-in-ggplot2
- 
-->

<!--# Expression for KEGG pathway of interest


TO ADD - barplot or similar summary of fold-change values for each enrichment 

-->

# Summary

- Set comparisons at the pathway level can be useful for understanding our enrichment results
- Venn diagrams are an easy way to compare between two sets
- We can create sophisticated and customized plots with enough persistance (and help from the internet)

<br/>
<br/>
<hr/>
| [Previous lesson](06-WebGestaltRGSEA.html) | [Top of this lesson](#top) | [Next lesson](08-analysis-summary.html) |
| :--- | :----: | ---: |