<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>GSEA with R</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link rel="shortcut icon" href="favicon-16x16.png" />
<!--
Favicon from built via
https://favicon.io/emoji-favicons/computer/
-->
<meta name="google-site-verification" content="t-Wgs-RXt8xi94RSoFamxoD2P2h-Hh7M_t_iOuxFZUo" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intro Functional Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="workshop-intro.html">Intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-functional-analysis-overview.html">Functional analysis and ORA</a>
    </li>
    <li>
      <a href="02-IntroToWebGestaltandORA.html">ORA with WebGestalt</a>
    </li>
    <li>
      <a href="03-WebGestaltRORA.html">ORA with R</a>
    </li>
    <li>
      <a href="04-gene-set-references.html">GO, KEGG, and friends</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="05-fcs-gsea-overview.html">Functional Analysis and GSEA</a>
    </li>
    <li>
      <a href="06-WebGestaltRGSEA.html">GSEA with R</a>
    </li>
    <li>
      <a href="07-advanced-visualizations.html">Advanced visualizations</a>
    </li>
    <li>
      <a href="08-analysis-summary.html">Analysis summary and next steps</a>
    </li>
  </ul>
</li>
<li>
  <a href="exercises.html">Exercises</a>
</li>
<li>
  <a href="workshop-wrap-up.html">Wrap up</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GSEA with R</h1>

</div>


<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>
<script>
  addClassKlippyTo("pre.r, pre.markdown, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<pre><code>## [1] &quot;/Users/cgates/git/workshop-intro-functional-analysis/source&quot;</code></pre>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<ul>
<li>Form the inputs for Gene Set Enrichment Analysis (GSEA) in
WebGestaltR from a differential expression analysis.</li>
<li>Use WebGestaltR to do a GSEA analysis.</li>
<li>Understand how to interpret the results of GSEA analysis.</li>
<li>Compare and contrast the results of two GSEA results.</li>
</ul>
</div>
<div id="getting-started" class="section level1">
<h1>Getting started</h1>
<p>The setup for GSEA is similar to ORA, but there are two key
differences:</p>
<ol style="list-style-type: decimal">
<li>We do not require a threshold for significance, and</li>
<li>We need to pass gene identifiers <strong>and</strong> the log
fold-change for the gene from the differential expression test.</li>
</ol>
<p>Again, we will take a look at the documentation for the
<code>WebGestaltR()</code> function, and note the difference in input
between ORA and GSEA.</p>
<pre class="r"><code># =========================================================================
# ORA with WebGestlatR
# =========================================================================

# -------------------------------------------------------------------------
# Load the libraries
library(WebGestaltR)
library(tidyverse)</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Look at the manual for the main function
?WebGestaltR</code></pre>
<p>We note from the documentation that <code>interestGene</code> “should
be an R data.frame object containing two columns: the gene list and the
corresponding scores.” We also note that we will <strong>not</strong>
make use of the <code>referenceGene</code> argument.</p>
<div id="read-in-diffex-results" class="section level2">
<h2>Read in diffex results</h2>
<p>We will demonstrate GSEA on a differential analysis from the
single-cell RNA-seq experiment we explored in the Introduction to
Single-cell Analysis <a
href="https://umich-brcf-bioinf.github.io/workshop-intro-single-cell/main/html/00A-OrientingOnScRNASeq.html#Consider_a_specific_scRNA-Seq_experiment">(link)</a>.
In particular, we perform GSEA on the differential expression results in
the pericyte cluster, comparing day 21 to day 7.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Load pseudo-bulk analysis for D21 vs D7 in pericyte cluster
isc_d21_v_d7 = read_csv(&#39;inputs/single_cell_de_results/de_pseudo_pericyte_D21_vs_D7.csv&#39;)</code></pre>
<pre><code>Rows: 23608 Columns: 6
── Column specification ──────────────────────────────────────────────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr (1): gene
dbl (5): p_val, avg_log2FC, pct.1, pct.2, p_val_adj

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Preview the result
head(isc_d21_v_d7)</code></pre>
<pre><code># A tibble: 6 × 6
      p_val avg_log2FC pct.1 pct.2 p_val_adj gene   
      &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;  
1 3.86e-232       2.91     1     1 1.02e-227 Cilp2  
2 5.63e-118       1.71     1     1 1.49e-113 Ltbp4  
3 1.85e- 95       1.29     1     1 4.91e- 91 Cd55   
4 2.60e- 87       1.64     1     1 6.88e- 83 Prss23 
5 3.02e- 87      -1.59     1     1 8.00e- 83 Nav3   
6 7.46e- 82       2.25     1     1 1.98e- 77 Gm42418</code></pre>
</div>
<div id="extracting-data-for-gsea" class="section level2">
<h2>Extracting data for GSEA</h2>
<p>The key columns to extract, as per the documentation, are the
<code>gene</code> and <code>avg_log2FC</code> columns. As before, we’ll
want to see how many gene symbols are NA and remove them.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# How many symbols are NA?
table(is.na(isc_d21_v_d7$gene))</code></pre>
<pre><code>
FALSE 
23608 </code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Filter out the NAs
isc_d21_v_d7 = isc_d21_v_d7 %&gt;% dplyr::filter(!is.na(gene))

# Verify we have fewer by noting dimension of resulting table
isc_d21_v_d7</code></pre>
<pre><code># A tibble: 23,608 × 6
       p_val avg_log2FC pct.1 pct.2 p_val_adj gene   
       &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;  
 1 3.86e-232      2.91      1     1 1.02e-227 Cilp2  
 2 5.63e-118      1.71      1     1 1.49e-113 Ltbp4  
 3 1.85e- 95      1.29      1     1 4.91e- 91 Cd55   
 4 2.60e- 87      1.64      1     1 6.88e- 83 Prss23 
 5 3.02e- 87     -1.59      1     1 8.00e- 83 Nav3   
 6 7.46e- 82      2.25      1     1 1.98e- 77 Gm42418
 7 3.19e- 81      1.16      1     1 8.44e- 77 Osbpl3 
 8 3.70e- 78      1.06      1     1 9.79e- 74 Lamb2  
 9 3.09e- 76      2.00      1     1 8.19e- 72 Wfdc1  
10 1.80e- 73      0.859     1     1 4.76e- 69 Abtb2  
# ℹ 23,598 more rows</code></pre>
<p>And now we can create a table with just the <code>gene</code> and
<code>avg_log2FC</code> columns with the <code>select()</code> function
from <code>dplyr</code>:</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Select gene and avg_log2FC columns
isc_d21_v_d7_gsea = isc_d21_v_d7 %&gt;%
    dplyr::select(gene, avg_log2FC)

# -------------------------------------------------------------------------
# Preview the table
head(isc_d21_v_d7_gsea)</code></pre>
<pre><code># A tibble: 6 × 2
  gene    avg_log2FC
  &lt;chr&gt;        &lt;dbl&gt;
1 Cilp2         2.91
2 Ltbp4         1.71
3 Cd55          1.29
4 Prss23        1.64
5 Nav3         -1.59
6 Gm42418       2.25</code></pre>
</div>
</div>
<div id="running-gsea" class="section level1">
<h1>Running GSEA</h1>
<p>As with the ORA, we will perform GSEA on the KEGG pathways and a
non-redundant version of the Gene Ontology Biological Process
genesets.</p>
<pre class="r"><code>isc_d21_v_d7_gsea_result = WebGestaltR(
    enrichMethod = &#39;GSEA&#39;,
    nThreads = 8,
    organism = &#39;mmusculus&#39;,
    enrichDatabase = c(&#39;pathway_KEGG&#39;, &#39;geneontology_Biological_Process_noRedundant&#39;),
    interestGene = isc_d21_v_d7_gsea,
    interestGeneType = &#39;genesymbol&#39;,
    fdrThr = 0.1,
    outputDirectory = &#39;./results&#39;,
    projectName = &#39;D21_v_D7_GSEA&#39;)</code></pre>
<pre><code>Loading the functional categories...
Loading the ID list...</code></pre>
<pre><code>Warning in dir.create(projectDir): &#39;./results/Project_D21_v_D7_GSEA&#39; already exists</code></pre>
<pre><code>Summarizing the uploaded ID list by GO Slim data...</code></pre>
<pre><code>Performing the enrichment analysis...</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00072</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00130</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00232</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00290</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00400</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00440</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00471</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00472</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00524</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00740</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00750</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00780</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00785</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu00920</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score does not contain minimum number of items in set for mmu04122</code></pre>
<pre><code>Warning in swGsea(inputDf, thresh_type = &quot;val&quot;, perms = perNum, min_set_size = minNum, : score has more than 500 items in set mmu01100</code></pre>
<pre><code>1000 permutations of score complete...</code></pre>
<pre><code>Begin affinity propagation...</code></pre>
<pre><code>Warning in .local(s, x = x, ...): algorithm did not converge; turn on details
and call plot() to monitor net similarity. Consider
increasing &#39;maxits&#39; and &#39;convits&#39;, and, if oscillations occur
also increasing damping factor &#39;lam&#39;.</code></pre>
<pre><code>End affinity propagation...
Begin weighted set cover...
End weighted set cover...
Generate the final report...
Results can be found in the ./results/Project_D21_v_D7_GSEA!</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# View the first few results
head(isc_d21_v_d7_gsea_result)</code></pre>
<pre><code>     geneSet              description
1   mmu04512 ECM-receptor interaction
2 GO:0007059   chromosome segregation
3   mmu03030          DNA replication
4   mmu03050               Proteasome
5   mmu03040              Spliceosome
6 GO:0006397          mRNA processing
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  link
1                                                                                                                                                                                                                                                                                                                                    http://www.kegg.jp/kegg-bin/show_pathway?mmu04512+12824+12845+12643+12841+21825+12840+21828+12839+12827+16779+20971+226519+12826+213119+12843+13138+16773+15530+12842+21827+15891+12834+16404+12833+11603+21826+319480+16398+12835+16776+81877+16419+14268+109700
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  http://amigo.geneontology.org/amigo/term/GO:0007059
3                                                                                                                                                                                                                                                                                                                                                                    http://www.kegg.jp/kegg-bin/show_pathway?mmu03030+19076+18968+18974+106344+68275+18972+19075+69724+68209+19687+18971+59001+72151+69263+17217+381760+67153+17220+17215+14156+19718+17218+68240+19891+17216+17219+18538+16881+66979
4                                                                                                                                                                                                                                                                                                                     http://www.kegg.jp/kegg-bin/show_pathway?mmu03050+19192+56436+16912+26440+66997+70247+19177+22123+20422+23996+19166+19179+19167+19184+19173+17463+19186+69077+26443+66413+23997+21762+19172+66537+103554+19171+26442+19181+19170+19188+19185+26445+26441+26446+26444+19175+16913
5 http://www.kegg.jp/kegg-bin/show_pathway?mmu03040+57905+20637+54633+66492+68879+20639+331401+72654+53317+320632+212880+66810+67039+68183+56031+72047+66354+56194+20624+60365+20222+22185+76522+81898+101943+66441+20643+230596+68092+13204+67465+66055+76938+68479+229279+20826+20638+68988+17149+56009+68011+27756+20630+75062+69878+107701+101214+56215+319322+66585+20382+19655+23983+73666+231889+107686+67959+67996+27366+51810+108014+20641+78651+66094+15381+108121+28000+68981+67332+15387+66101+53607+53817+76936+19656+20462+67678+68816+66125+66373+110809+50783+20383+15382+225027+21681
6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  http://amigo.geneontology.org/amigo/term/GO:0006397
  enrichmentScore normalizedEnrichmentScore pValue         FDR size                                    plotPath
1       0.7715357                  2.101835      0 0.000000000   82   ./Project_D21_v_D7_GSEA_GSEA/mmu04512.png
2      -0.6280825                 -1.875460      0 0.000000000  297 ./Project_D21_v_D7_GSEA_GSEA/GO_0007059.png
3      -0.8525372                 -1.891195      0 0.000000000   35   ./Project_D21_v_D7_GSEA_GSEA/mmu03030.png
4      -0.8164866                 -1.913877      0 0.000000000   46   ./Project_D21_v_D7_GSEA_GSEA/mmu03050.png
5      -0.7258237                 -1.949957      0 0.000000000  128   ./Project_D21_v_D7_GSEA_GSEA/mmu03040.png
6      -0.6127408                 -1.863062      0 0.000182216  407 ./Project_D21_v_D7_GSEA_GSEA/GO_0006397.png
  leadingEdgeNum
1             34
2            152
3             29
4             37
5             86
6            203
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     leadingEdgeId
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  12824;12845;12643;12841;21825;12840;21828;12839;12827;16779;20971;226519;12826;213119;12843;13138;16773;15530;12842;21827;15891;12834;16404;12833;11603;21826;319480;16398;12835;16776;81877;16419;14268;109700
2                                                                                                                                                                                                                                                                                                                               26951;72124;217232;208718;211651;68549;51792;214901;56742;67141;219114;23834;72107;215387;58186;70385;223970;71175;17535;52683;78610;229776;12617;66934;67166;57261;71988;67139;230936;18005;12235;20479;381293;12189;71878;78658;218914;68501;66156;67849;110147;70012;74117;100710;66385;12704;21969;54392;226151;21974;19357;72415;18226;12448;20892;72787;83815;16571;66591;66713;52206;22284;54141;66977;224727;17120;13605;66578;13018;69912;13990;224023;73804;72117;67177;20815;66468;12236;72155;80912;229524;56292;21951;16211;75801;53869;52696;12387;56150;210711;214444;108912;13006;67203;67052;108907;83560;226026;21749;66617;268465;69716;69736;237911;70099;66140;71846;70799;12540;71819;20842;12237;68298;22680;108000;69928;27050;22589;26886;51944;100502766;100088;12447;75430;20877;12034;110033;51810;16319;229841;17345;71514;66311;14211;68523;103733;268697;18817;26914;26934;66440;67037;66442;68250;52276;22196;21973;107995;19384;59126;11740;11799
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  19076;18968;18974;106344;68275;18972;19075;69724;68209;19687;18971;59001;72151;69263;17217;381760;67153;17220;17215;14156;19718;17218;68240;19891;17216;17219;18538;16881;66979
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   19192;56436;16912;26440;66997;70247;19177;22123;20422;23996;19166;19179;19167;19184;19173;17463;19186;69077;26443;66413;23997;21762;19172;66537;103554;19171;26442;19181;19170;19188;19185;26445;26441;26446;26444;19175;16913
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               57905;20637;54633;66492;68879;20639;331401;72654;53317;320632;212880;66810;67039;68183;56031;72047;66354;56194;20624;60365;20222;22185;76522;81898;101943;66441;20643;230596;68092;13204;67465;66055;76938;68479;229279;20826;20638;68988;17149;56009;68011;27756;20630;75062;69878;107701;101214;56215;319322;66585;20382;19655;23983;73666;231889;107686;67959;67996;27366;51810;108014;20641;78651;66094;15381;108121;28000;68981;67332;15387;66101;53607;53817;76936;19656;20462;67678;68816;66125;66373;110809;50783;20383;15382;225027;21681
6 234723;22380;72198;66070;171170;28035;20227;56403;66185;14300;20924;13211;51786;54723;233900;229709;70225;238831;19826;57905;20637;54633;209003;67010;66492;74111;68879;69131;20639;69562;69241;331401;65105;53317;218543;320632;100715;212880;68927;54188;381802;53890;268859;72692;224432;66810;72290;83410;18747;67039;66660;67242;237221;214572;101471;67178;20901;68183;230753;56031;66880;107951;231386;107817;66354;67755;20817;78372;56194;22687;53861;27632;20624;60365;12192;100336;68035;224903;60532;66177;74213;18789;22185;66603;219158;20595;76522;51796;67487;101739;22083;386612;232314;16549;101943;12455;66441;66637;20643;74737;69305;19712;230596;83701;104721;24128;68092;59092;20815;13204;54196;67465;116748;229700;66055;76938;19663;68479;229279;18458;20638;68988;12457;17149;27374;21841;56009;69269;68011;27756;20630;66077;75062;14007;69878;72935;230908;67684;107701;56215;319322;66585;20382;19655;20220;23983;19383;73666;432508;66078;231889;68991;107686;75956;53610;67959;67996;27366;51810;108014;20641;20218;78651;66094;15381;269966;70248;108121;59013;28000;71514;68981;67332;15387;71973;18148;68219;66101;268697;53607;228410;268420;53817;76936;19656;20462;67678;68816;98758;66125;66373;110809;19652;50783;20383;53379;15382;22608;225027;11758;21681;66704;20492
                                     database
1                                pathway_KEGG
2 geneontology_Biological_Process_noRedundant
3                                pathway_KEGG
4                                pathway_KEGG
5                                pathway_KEGG
6 geneontology_Biological_Process_noRedundant
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         userId
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Agrn;Chad;Col1a1;Col1a2;Col2a1;Col4a1;Col4a2;Col6a1;Col6a2;Col6a3;Col9a1;Col9a2;Col9a3;Comp;Dag1;Fn1;Hspg2;Ibsp;Itga1;Itga10;Itga11;Itga2;Itga7;Itgb5;Lama2;Lama5;Lamb2;Lamc1;Sdc4;Thbs1;Thbs2;Thbs3;Thbs4;Tnxb
2                                                                                                                                                                                                                                                                                                                                                  4930447C04Rik;Actr2;Actr3;Anapc11;Anapc15;Anapc4;Arl8b;Atrx;Aurkb;Bag6;Birc5;Brca1;Brd4;Brip1;Bub1;Bub1b;Bub3;Ccnb1;Ccne1;Ccne2;Cdc14a;Cdc20;Cdc26;Cdc27;Cdc42;Cdc6;Cdca2;Cdca5;Cdca8;Cdk5rap2;Cdt1;Cenpc1;Cenpe;Cenpf;Cenph;Cenpn;Cenpq;Cenps;Cenpw;Cenpx;Cep192;Cep85;Chtf18;Ciao2a;Ciao2b;Cit;Ctcf;Ctnnb1;Dis3l2;Dscc1;Dsn1;Ect2;Ehmt2;Eme1;Esco2;Fam83d;Fancd2;Fbxo5;H2afy;Hnrnpu;Incenp;Kif14;Kif22;Kif23;Kif2c;Kif4;Kifc1;Klhl22;Knstrn;Kpnb1;Mad1l1;Mad2l1;Mad2l1bp;Mcmbp;Mis12;Mis18a;Mki67;Mre11a;Msto1;Naa10;Naa50;Ncapd2;Ncapd3;Ncapg;Ncaph;Ncaph2;Ndc1;Ndc80;Nde1;Nek2;Nek6;Nipbl;Nsmce2;Ntmt1;Nuf2;Nup37;Nup43;Nup62;Nusap1;Pds5b;Phb2;Phf13;Plk1;Pmf1;Ppp1r7;Ppp2r1a;Psrc1;Pum1;Rab11a;Racgap1;Rad18;Rad21;Ran;Rcc1;Rmi2;Rps3;Seh1l;Sfpq;Sgo1;Sgo2a;Ska1;Ska2;Ska3;Slc25a5;Slf2;Smarcad1;Smc2;Smc3;Smc4;Smc5;Spag5;Spc25;Spdl1;Srpk1;Stag1;Syce2;Terf1;Tex14;Tnks;Top1;Top2a;Top2b;Trip13;Tubg1;Ube2i;Usp9x;Uvrag;Vps4b;Wapl;Zfp207;Zw10;Zwint
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         Fen1;Lig1;Mcm2;Mcm3;Mcm4;Mcm5;Mcm6;Mcm7;Pcna;Pola1;Pold1;Pold2;Pole2;Pole3;Pole4;Prim1;Prim2;Rfc1;Rfc2;Rfc3;Rfc4;Rfc5;Rnaseh2a;Rnaseh2b;Rnaseh2c;Rpa1;Rpa2;Rpa3;Ssbp1
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Adrm1;Pomp;Psma1;Psma2;Psma3;Psma4;Psma5;Psma6;Psma7;Psmb1;Psmb10;Psmb2;Psmb3;Psmb4;Psmb5;Psmb6;Psmb7;Psmb8;Psmb9;Psmc1;Psmc2;Psmc4;Psmc5;Psmd1;Psmd11;Psmd12;Psmd13;Psmd2;Psmd3;Psmd4;Psmd6;Psmd7;Psme1;Psme2;Psme3;Psme4;Sem1
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    Acin1;Alyref;Alyref2;Bcas2;Bud31;Ccdc12;Ddx39b;Ddx42;Ddx46;Dhx15;Eftud2;Hnrnpa1;Hnrnpa3;Hnrnpc;Hnrnpk;Hnrnpm;Hnrnpu;Isy1;Lsm2;Lsm3;Lsm4;Lsm5;Lsm6;Lsm7;Lsm8;Magoh;Magohb;Ncbp2;Pcbp1;Phf5a;Plrg1;Ppie;Ppih;Ppil1;Pqbp1;Prpf19;Prpf31;Prpf38a;Prpf40a;Prpf6;Puf60;Rbm17;Rbm22;Rbm25;Rbm8a;Rbmx;Rbmxl1;Sf3a1;Sf3a2;Sf3a3;Sf3b1;Sf3b2;Sf3b3;Sf3b4;Sf3b5;Sf3b6;Snrnp200;Snrnp40;Snrnp70;Snrpa;Snrpa1;Snrpb;Snrpb2;Snrpc;Snrpd1;Snrpd2;Snrpd3;Snrpe;Snrpf;Snrpg;Snu13;Snw1;Srsf1;Srsf2;Srsf3;Srsf6;Srsf7;Srsf9;Thoc2;Thoc3;Tra2a;Tra2b;Txnl4a;U2af1;U2af2;Zmat2
6 Acin1;Ahcyl1;Alkbh5;Alyref;Alyref2;Arl6ip4;Aurkaip1;Bcas2;Bud31;Ccar2;Ccnb1;Ccnt1;Cdk12;Cdk13;Cdk9;Celf2;Cpsf2;Cpsf4;Cpsf6;Cstf2t;Cstf3;Ctr9;Cwc15;Dazap1;Dcps;Ddx1;Ddx39b;Ddx41;Ddx46;Ddx47;Dhx15;Dhx40;Dhx9;Eftud2;Frg1;Gemin2;Gemin6;Gemin8;Hnrnpa1;Hnrnpa2b1;Hnrnpa3;Hnrnpc;Hnrnpf;Hnrnph1;Hnrnpk;Hnrnpll;Hnrnpm;Hnrnpu;Isy1;Jmjd6;Khdrbs1;Khsrp;Lsm10;Lsm11;Lsm2;Lsm3;Lsm4;Lsm5;Lsm6;Lsm7;Lsm8;Luc7l3;Magoh;Magohb;Mbnl3;Mtrex;Ncbp2;Nelfe;Noct;Nono;Npm1;Nudt21;Nup98;Pabpc1;Pabpn1;Papola;Pcbp1;Pcbp4;Pcf11;Phf5a;Phrf1;Plrg1;Polr2d;Ppie;Ppih;Ppil1;Ppil3;Ppp1r8;Ppp4r2;Ppwd1;Pqbp1;Prdx6;Prkaca;Prmt5;Prmt7;Prpf19;Prpf31;Prpf38a;Prpf40a;Prpf6;Psip1;Ptcd2;Puf60;Raly;Rbfox1;Rbm15;Rbm17;Rbm19;Rbm22;Rbm25;Rbm26;Rbm3;Rbm42;Rbm4b;Rbm7;Rbm8a;Rbmx;Rbmx2;Rbmxl1;Rbpms;Rbpms2;Rest;Rnf40;Rnps1;Rsrc1;Safb;Sap18;Sart1;Sart3;Scaf4;Scnm1;Sf3a1;Sf3a3;Sf3b2;Sf3b3;Sf3b4;Sf3b5;Sf3b6;Sfpq;Slbp;Sltm;Smn1;Snrnp200;Snrnp25;Snrnp40;Snrnp70;Snrpa;Snrpa1;Snrpb;Snrpb2;Snrpc;Snrpd1;Snrpd2;Snrpd3;Snrpe;Snrpf;Snrpg;Snw1;Srek1;Srpk1;Srpk2;Srrm1;Srrm2;Srrt;Srsf1;Srsf2;Srsf3;Srsf6;Srsf7;Srsf9;Ssu72;Strap;Supt5;Syncrip;Tardbp;Tent2;Tfip11;Thoc2;Thoc3;Thoc6;Thrap3;Tia1;Tra2b;Tsen15;Tsen2;Tsen34;Txnl4a;Txnl4b;U2af1;U2af2;Ubl5;Usp39;Virma;Wbp4;Wtap;Xrn2;Ybx1;Ythdc1;Zfp36l1;Zmat2;Zmat5;Zpr1;Zranb2</code></pre>
<p>The columns of the GSEA results are quite similar to the ORA results,
with the following exceptions:</p>
<ul>
<li><code>enrichmentScore</code>:</li>
<li><code>normalizedEnrichmentScore</code>:</li>
<li><code>leadingEdgeNum</code>:</li>
</ul>
<p>TODO: Here is where we’ll actually talk more about the biological
relevance of the results.</p>
</div>
<div id="results-of-gsea" class="section level1">
<h1>Results of GSEA</h1>
<p>As with the ORA results, the GSEA method outputs similar results
tables, gene identifier mappings, and HTML reports. Additionally, for
each pathway with <code>FDR &lt; fdrThr</code>, a GSEA enrichment plot
is automatically generated. The following is an example:</p>
<p>TODO: Include example GSEA plot.</p>
</div>
<div id="exercise" class="section level1">
<h1>Exercise</h1>
</div>
<div id="comparing-enrichment-results" class="section level1">
<h1>Comparing enrichment results</h1>
<p>Now that we have two GSEA results from related differential
expression results, we may want to know:</p>
<ul>
<li>What genesets are in common?</li>
<li>What genesets are unique?</li>
<li>Are there any insights in the pathways at one time point versus
another?</li>
</ul>
<p>A first approach to these questions might be to take the GSEA results
from both comparisons and join them on the geneset identifier.</p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
</div>
<div id="questions" class="section level1">
<h1>Questions</h1>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"><a href="05-fcs-gsea-overview.html">Previous
lesson</a></th>
<th align="center"><a href="#top">Top of this lesson</a></th>
<th align="right"><a href="07-advanced-visualizations.html">Next
lesson</a></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
