<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Advanced Visualizations II</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/clipboard-1.7.1/clipboard.min.js"></script>
<link href="site_libs/primer-tooltips-1.4.0/build.css" rel="stylesheet" />
<link href="site_libs/klippy-0.0.0.9500/css/klippy.min.css" rel="stylesheet" />
<script src="site_libs/klippy-0.0.0.9500/js/klippy.min.js"></script>
<link rel="shortcut icon" href="favicon-16x16.png" />
<!--
Favicon from built via
https://favicon.io/emoji-favicons/computer/
-->
<meta name="google-site-verification" content="t-Wgs-RXt8xi94RSoFamxoD2P2h-Hh7M_t_iOuxFZUo" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intro Functional Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="workshop-intro.html">Intro</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 1
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-overview.pdf">Functional analysis and ORA</a>
    </li>
    <li>
      <a href="02-IntroToWebGestaltandORA.html">ORA with WebGestalt</a>
    </li>
    <li>
      <a href="03-WebGestaltRORA.html">ORA with R</a>
    </li>
    <li>
      <a href="07A-advanced-visualizations.html">Advanced visualizations I</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Day 2
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ifun-gsea-overview.pdf">Functional Analysis and GSEA</a>
    </li>
    <li>
      <a href="06-WebGestaltRGSEA.html">GSEA with R</a>
    </li>
    <li>
      <a href="gene-set-references.pdf">GO, KEGG, and friends</a>
    </li>
    <li>
      <a href="07B-advanced-visualizations.html">Advanced visualizations II</a>
    </li>
    <li>
      <a href="08-analysis-summary.html">Analysis summary</a>
    </li>
  </ul>
</li>
<li>
  <a href="workshop-wrap-up.html">Wrap up</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Advanced Visualizations II</h1>

</div>


<style type="text/css">

body, td {
   font-size: 16px;
}
code.r{
  font-size: 12px;
}
pre {
  font-size: 12px
}

</style>
<script>
  addClassKlippyTo("pre.r, pre.markdown, pre.bash");
  addKlippy('right', 'top', 'auto', '1', 'Copy code', 'Copied!');
</script>
<p><img src="images/wayfinder/wayfinder-07B.png"
alt="workshop wayfinder" /></p>
<div id="objectives" class="section level1">
<h1>Objectives</h1>
<ul>
<li>Compare significant pathways between multiple enrichment
results</li>
<li>Highlight some interesting pathways from the meta-analysis</li>
</ul>
</div>
<div id="getting-started" class="section level1">
<h1>Getting started</h1>
<!-- # library(VennDiagram) # missing library -->
<pre class="r"><code># =========================================================================
# Advanced Visualizations II
# =========================================================================

# -------------------------------------------------------------------------
# Load additional libraries
library(ComplexHeatmap)
library(ggVennDiagram)

# -------------------------------------------------------------------------
# Check current working directory
getwd()</code></pre>
<pre><code>[1] &quot;/Users/cgates/git/workshop-intro-functional-analysis/source&quot;</code></pre>
<p>We should have the GSEA results in our session but we can read in the
results from the table output to file by <code>WebGestaltR</code> for
the results from the Day21 vs Day7 and Day7 vs Day 0 pericyte
single-cell RNA-seq comparisons.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Read in results for the enrichment for each set of timpoint comparisons
isc_d21_v_d7_gsea_result = read_delim(&#39;results/Project_D21_v_D7_GSEA_KEGG/enrichment_results_D21_v_D7_GSEA_KEGG.txt&#39;)</code></pre>
<pre><code>Rows: 37 Columns: 12
── Column specification ──────────────────────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (6): geneSet, description, link, plotPath, leadingEdgeId, userId
dbl (6): enrichmentScore, normalizedEnrichmentScore, pValue, FDR, size, leadingEdgeNum

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>isc_d7_v_d0_gsea_result = read_delim(&#39;results/Project_D7_v_D0_GSEA_KEGG//enrichment_results_D7_v_D0_GSEA_KEGG.txt&#39;)</code></pre>
<pre><code>Rows: 16 Columns: 12
── Column specification ──────────────────────────────────────────────────────────────────────────
Delimiter: &quot;\t&quot;
chr (6): geneSet, description, link, plotPath, leadingEdgeId, userId
dbl (6): enrichmentScore, normalizedEnrichmentScore, pValue, FDR, size, leadingEdgeNum

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>head(isc_d21_v_d7_gsea_result); head(isc_d7_v_d0_gsea_result)</code></pre>
<pre><code># A tibble: 6 × 12
  geneSet  description  link  enrichmentScore normalizedEnrichment…¹ pValue     FDR  size plotPath
  &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt;           &lt;dbl&gt;                  &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   
1 mmu04512 ECM-recepto… http…           0.772                   2.09      0 0          82 ./Proje…
2 mmu03050 Proteasome   http…          -0.816                  -1.89      0 0          46 ./Proje…
3 mmu03030 DNA replica… http…          -0.853                  -1.90      0 0          35 ./Proje…
4 mmu03040 Spliceosome  http…          -0.726                  -1.96      0 0         128 ./Proje…
5 mmu04657 IL-17 signa… http…          -0.690                  -1.78      0 0.00146    86 ./Proje…
6 mmu05012 Parkinson d… http…          -0.651                  -1.77      0 0.00155   125 ./Proje…
# ℹ abbreviated name: ¹​normalizedEnrichmentScore
# ℹ 3 more variables: leadingEdgeNum &lt;dbl&gt;, leadingEdgeId &lt;chr&gt;, userId &lt;chr&gt;</code></pre>
<pre><code># A tibble: 6 × 12
  geneSet  description  link  enrichmentScore normalizedEnrichment…¹ pValue     FDR  size plotPath
  &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt;           &lt;dbl&gt;                  &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   
1 mmu03010 Ribosome     http…           0.791                   2.42      0 0         129 ./Proje…
2 mmu03030 DNA replica… http…           0.766                   1.90      0 0.00423    35 ./Proje…
3 mmu05134 Legionellos… http…           0.676                   1.81      0 0.0137     56 ./Proje…
4 mmu05143 African try… http…           0.732                   1.78      0 0.0154     33 ./Proje…
5 mmu05012 Parkinson d… http…           0.563                   1.73      0 0.0204    125 ./Proje…
6 mmu00190 Oxidative p… http…           0.572                   1.74      0 0.0216    115 ./Proje…
# ℹ abbreviated name: ¹​normalizedEnrichmentScore
# ℹ 3 more variables: leadingEdgeNum &lt;dbl&gt;, leadingEdgeId &lt;chr&gt;, userId &lt;chr&gt;</code></pre>
<p><br></p>
<p>Instead of comparing gene sets between enriched terms from a single
set of results, we can instead compare enrichments between two different
comparisons. Since the KEGG ids and descriptions are organized into
columns of the results tables we don’t need to do as much processing to
compare between the enriched pathways.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Create list of enriched KEGG pathways 
KEGG_results &lt;- c()
KEGG_results[[&quot;d7_v_d0&quot;]] = isc_d7_v_d0_gsea_result$description
KEGG_results[[&quot;d21_v_d7&quot;]] = isc_d21_v_d7_gsea_result$description</code></pre>
<p>Once we’ve created the lists of enriched pathways, we can use
operations like <code>union</code> and <code>intersect</code> to compare
between the sets.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Identify total number of unique pathways enriched across both timepoints
union(KEGG_results[[&quot;d7_v_d0&quot;]], KEGG_results[[&quot;d21_v_d7&quot;]]) %&gt;% length()</code></pre>
<pre><code>[1] 40</code></pre>
<pre class="r"><code># -------------------------------------------------------------------------
# Identify number of pathways shared between timepoints
intersect(KEGG_results[[&quot;d7_v_d0&quot;]], KEGG_results[[&quot;d21_v_d7&quot;]]) %&gt;% length()</code></pre>
<pre><code>[1] 13</code></pre>
<p>Since the number of sets in one of our comparisons exceeds the limit
allowed for <code>ComplexHeatmap</code>’s version of an upSet plot
(which is <code>31</code>), we’ll create a venn diagram comparing the
number of shared/unique pathways for each timepoint using the
<code>ggVennDiagram</code> option, which is <a
href="https://www.datanovia.com/en/blog/beautiful-ggplot-venn-diagram-with-r">one
of many options available to create this kind of plot</a>, instead.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Make simple plot to compare KEGG pathways enriched for each timepoint
KEGG_venn &lt;- ggVennDiagram(KEGG_results, label_alpha = 0, 
                           label_size = 6, set_size = 6,
                           ) +
  ggplot2::scale_fill_gradient(low=&quot;royalblue&quot;,high = &quot;yellow&quot;) +
  coord_flip()</code></pre>
<pre><code>Coordinate system already present. Adding new coordinate system, which will replace the existing
one.</code></pre>
<pre class="r"><code>KEGG_venn</code></pre>
<p><img src="fig/02-path_Venn-1.png" width="816" style="display: block; margin: auto;" />
<br></p>
<p>Then we can output the plot to file:</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Output UpSet plot to file
png(file = &quot;./results/figures/Pericytes_KEGG_vennDiagram.png&quot;, width = 600, height = 400)
  KEGG_venn # draw the plot
dev.off() # Close the graphics device</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div id="detailed-comparison-of-kegg-enrichments"
class="section level1">
<h1>Detailed comparison of KEGG enrichments</h1>
<p>From the Venn diagram, we have a summary of the number of KEGG
pathways that are shared or uniquely enriched for the two sets of
comparisons from the pericyte population but that doesn’t tell us the
magnitudes of those enrichments or if they were in the same direction or
opposite direction between the earlier timepoint comparison and later
timepoint comparison.</p>
<!-- ideally add EnrichR and/or ClusterProfiler packages since have improved visualizations compared to  -->
<pre class="r"><code># -------------------------------------------------------------------------
# Create &#39;tidy&#39; version  of Day 7 vs Day 0 GSEA results
isc_d7_v_d0_gsea_simplified &lt;- isc_d7_v_d0_gsea_result %&gt;% 
  select(geneSet, description, normalizedEnrichmentScore, size, FDR) %&gt;% 
  pivot_longer(cols = !c(geneSet, description, FDR, size)) %&gt;% 
  mutate(timepoint = &quot;D7_v_D0&quot;, .before = 1)

# -------------------------------------------------------------------------
# Create &#39;tidy&#39; version  of Day 21 vs Day 7 GSEA results
isc_d21_v_d7_gsea_simplified &lt;- isc_d21_v_d7_gsea_result %&gt;% 
  select(geneSet, description, normalizedEnrichmentScore, size, FDR) %&gt;% 
  pivot_longer(cols = !c(geneSet, description, FDR, size)) %&gt;% 
  mutate(timepoint = &quot;D21_v_D7&quot;, .before = 1)

# -------------------------------------------------------------------------
# Combine &#39;tidy&#39; results into single table
KEGG_results_combined &lt;- rbind(isc_d7_v_d0_gsea_simplified, isc_d21_v_d7_gsea_simplified) 
KEGG_results_combined &lt;- KEGG_results_combined %&gt;% 
  mutate(timepoint = factor(timepoint, levels=c(&quot;D7_v_D0&quot;, &quot;D21_v_D7&quot;)),
         direction = factor(ifelse(value &lt; 0, &quot;negative NES&quot;, &quot;positive NES&quot;)),
         &quot;positive NES&quot;, &quot;negative NES&quot;) # fix levels &amp; add column for direction

# -------------------------------------------------------------------------
# Check table
str(KEGG_results_combined)</code></pre>
<pre><code>tibble [53 × 10] (S3: tbl_df/tbl/data.frame)
 $ timepoint     : Factor w/ 2 levels &quot;D7_v_D0&quot;,&quot;D21_v_D7&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ geneSet       : chr [1:53] &quot;mmu03010&quot; &quot;mmu03030&quot; &quot;mmu05134&quot; &quot;mmu05143&quot; ...
 $ description   : chr [1:53] &quot;Ribosome&quot; &quot;DNA replication&quot; &quot;Legionellosis&quot; &quot;African trypanosomiasis&quot; ...
 $ size          : num [1:53] 129 35 56 33 125 115 86 46 111 94 ...
 $ FDR           : num [1:53] 0 0.00423 0.01371 0.01542 0.02039 ...
 $ name          : chr [1:53] &quot;normalizedEnrichmentScore&quot; &quot;normalizedEnrichmentScore&quot; &quot;normalizedEnrichmentScore&quot; &quot;normalizedEnrichmentScore&quot; ...
 $ value         : num [1:53] 2.42 1.9 1.81 1.78 1.73 ...
 $ direction     : Factor w/ 2 levels &quot;negative NES&quot;,..: 2 2 2 2 2 2 2 2 2 2 ...
 $ &quot;positive NES&quot;: chr [1:53] &quot;positive NES&quot; &quot;positive NES&quot; &quot;positive NES&quot; &quot;positive NES&quot; ...
 $ &quot;negative NES&quot;: chr [1:53] &quot;negative NES&quot; &quot;negative NES&quot; &quot;negative NES&quot; &quot;negative NES&quot; ...</code></pre>
<p>Now that we have a nice “tidy” version of the enrichment results, we
can use it to create a barplot that summarizes the enrichment scores for
each timepoint.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Graph simple KEGG pathway enrichments for both timepoints
KEGG_comparison &lt;- ggplot(KEGG_results_combined, 
                          aes(y=value, x=reorder(description, value))) + 
  geom_bar(position=&quot;dodge&quot;, stat=&quot;identity&quot;) +
  facet_wrap(~timepoint) +
  xlab(&quot;&quot;) + coord_flip()
KEGG_comparison</code></pre>
<p><img src="fig/02-plot_KEGG_simple-1.png" width="816" style="display: block; margin: auto;" /></p>
<p><br></p>
<p>While this barplot has some interesting patterns, it’s a bit hard to
read and isn’t very pretty.</p>
<p>Let’s create a modified version that filters our enrichment results
and adds attributes to the plot to show some key metrics, including the
direction and magnitude of the normalized enrichment score, the
significance, and the number of genes in the pathway (size).</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Subset to results that pass more stringent results
KEGG_results_strict &lt;- KEGG_results_combined %&gt;% filter(FDR &lt; 0.025 &amp; size &gt; 60)

# -------------------------------------------------------------------------
# Graph simple KEGG pathway enrichments for both timepoints
KEGG_comparison &lt;- ggplot(KEGG_results_strict, aes(y=value, x=reorder(description, value))) + 
  geom_segment(aes(y = 0, x = description, yend = value, xend = description),
               color = ifelse(KEGG_results_strict$direction %in% c(&quot;positive NES&quot;), &quot;red&quot;, &quot;blue&quot;),
               linewidth = 1) +
  geom_point(aes(color=FDR, size = size), stat=&quot;identity&quot;) +
  scale_color_continuous(low = &quot;black&quot;, high = &quot;grey&quot;) +
  facet_wrap(~timepoint) +
  geom_hline(yintercept=0, linetype=&quot;dashed&quot;, color = &quot;lightgrey&quot;) +
  labs(title=&quot;KEGG enrichments in pericytes&quot;) +
  xlab(&quot;&quot;) + theme_bw() +
  coord_flip()
KEGG_comparison</code></pre>
<p><img src="fig/02-plot_KEGG_complex-1.png" width="816" style="display: block; margin: auto;" />
<br></p>
<p>What we created here is called a “lollipop” plot, which we’ll save to
file.</p>
<pre class="r"><code># -------------------------------------------------------------------------
# Output UpSet plot to file
png(file = &quot;./results/figures/Pericytes_KEGG_lollipopComparison.png&quot;, width = 700, height = 600, res = 100)
  KEGG_comparison # draw the plot
dev.off() # Close the graphics device</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<p><br></p>
<p>From the comparison of enriched KEGG pathways between the timepoints,
one of the results that jumps out is the the positive enrichment score
for “IL-17 signaling pathway” at the early time points comparison and
the negative enrichment score at the later time points comparison. IL-17
signalling has a <a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC11129447/">role in wound
repair</a> so it’s activation at the early time point and suppression at
the later time point fits with our general expectations but might be
interesting depending on what’s known about IL-17 signaling in
pericytes.</p>
<p>Another result that jumps out is the enrichment only at the later
timepoint of “ECM-receptor interaction”, which has a <a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC9326521/">role in wound
healing</a> but also <a
href="https://pmc.ncbi.nlm.nih.gov/articles/PMC7264100/">bone
formation</a>.</p>
<p><br></p>
<!-- How did we decide to make this kind of plot and what did we do to create it? Add steps of search/iteration?
Links:
- https://r-graph-gallery.com/48-grouped-barplot-with-ggplot2.html
- https://r-charts.com/part-whole/diverging-bar-chart-ggplot2/
- https://www.tutorialspoint.com/ggplot2/ggplot2_diverging_charts.html
- https://yulab-smu.top/biomedical-knowledge-mining-book/enrichplot.html
- https://stackoverflow.com/questions/15116081/controlling-order-of-facet-grid-facet-wrap-in-ggplot2
- https://r-graph-gallery.com/304-highlight-a-group-in-lollipop.html
- https://ggplot2-book.org/scales-colour.html#sec-binned-colour
- https://www.sthda.com/english/wiki/ggplot2-colors-how-to-change-colors-automatically-and-manually
- https://stackoverflow.com/questions/15116081/controlling-order-of-facet-grid-facet-wrap-in-ggplot2
- 
-->
<!--# Expression for KEGG pathway of interest


TO ADD - barplot or similar summary of fold-change values for each enrichment 

-->
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<ul>
<li>Set comparisons at the pathway level can be useful for understanding
our enrichment results</li>
<li>Venn diagrams are an easy way to compare between two sets</li>
<li>We can create sophisticated and customized plots with enough
persistance (and help from the internet)</li>
</ul>
<br/> <br/>
<hr/>
<table style="width:100%;">
<colgroup>
<col width="28%" />
<col width="42%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"><a href="06-WebGestaltRGSEA.html">Previous
lesson</a></th>
<th align="center"><a href="#top">Top of this lesson</a></th>
<th align="right"><a href="08-analysis-summary.html">Next
lesson</a></th>
</tr>
</thead>
<tbody>
</tbody>
</table>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3,h4",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
